<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-04-10" />

<title>Analysis of Greedy Initialization for PBMC Purified Analysis</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">log1p_experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eweine/log1p_experiments">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Analysis of Greedy Initialization for PBMC
Purified Analysis</h1>
<h4 class="date">2024-04-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>log1p_experiments/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240402code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240402)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240402code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240402)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/log1p_experiments/output/pbmc_purified_25K_greedy_log1p_fit.rds
</td>
<td style="text-align:left;">
output/pbmc_purified_25K_greedy_log1p_fit.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Documents/log1p_experiments/output/pbmc_purified_25K_greedy_init_log1p_fit.rds
</td>
<td style="text-align:left;">
output/pbmc_purified_25K_greedy_init_log1p_fit.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Documents/log1p_experiments/output/pbmc_purified_25K_frob_nmf_init_log1p_fit.rds
</td>
<td style="text-align:left;">
output/pbmc_purified_25K_frob_nmf_init_log1p_fit.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstree950f789f2b1c3eb293a8e98467f3d34cb511defatargetblank950f789a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/eweine/log1p_experiments/tree/950f789f2b1c3eb293a8e98467f3d34cb511defa" target="_blank">950f789</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstree950f789f2b1c3eb293a8e98467f3d34cb511defatargetblank950f789a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/eweine/log1p_experiments/tree/950f789f2b1c3eb293a8e98467f3d34cb511defa" target="_blank">950f789</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/figure/

Unstaged changes:
    Modified:   analysis/greedy_init.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/greedy_init.Rmd</code>) and HTML
(<code>docs/greedy_init.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/950f789f2b1c3eb293a8e98467f3d34cb511defa/analysis/greedy_init.Rmd" target="_blank">950f789</a>
</td>
<td>
Eric Weine
</td>
<td>
2024-04-10
</td>
<td>
added greedy fit analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/950f789f2b1c3eb293a8e98467f3d34cb511defa/docs/greedy_init.html" target="_blank">950f789</a>
</td>
<td>
Eric Weine
</td>
<td>
2024-04-10
</td>
<td>
added greedy fit analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>Here, I analyzed the PBMC purified dataset using three different
methods: (1) Poisson NMF (as implemented in fastTopics), (2) log1p
transformed frobenius NMF, and (3) Poisson NMF with a log1p link
function.</p>
<pre class="r"><code>library(dplyr)
library(tidyr)
library(purrr)
library(stringr)</code></pre>
<pre class="r"><code>load(&quot;~/Documents/data/fastglmpca/raw_data/pbmc_purified.RData&quot;)
set.seed(1)
sz &lt;- Matrix::rowSums(counts) / mean(Matrix::rowSums(counts))
counts &lt;- counts[, Matrix::colSums(counts) &gt;= 100]
genes_map &lt;- genes$ensembl
names(genes_map) &lt;- genes$symbol

genes_map_inv &lt;- genes$symbol
names(genes_map_inv) &lt;- genes$ensembl</code></pre>
<p>Below is the code to run the log1p link Poisson NMF factorization
method greedily.</p>
<pre class="r"><code>library(Matrix)
set.seed(1)
log1p_greedy_fit &lt;- fit_factor_model_log1p_quad_approx_sparse_greedy(
  Y = counts,
  K = 25,
  iter_per_factor = 8,
  approx_range = c(0, 1.25),
  s = as.vector(sz)
)</code></pre>
<p>Below is the code to run the log1p link Poisson NMF factorization
method with a greedy initialization.</p>
<pre class="r"><code>set.seed(1)
log1p_fit_init_greedy &lt;- fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = 25,
  maxiter = 50,
  approx_range = c(0, 1.25),
  s = as.vector(sz),
  init_U = log1p_fit$U,
  init_V = log1p_fit$V
)</code></pre>
<p>Finally, below below is the code to run the log1p link Poisson NMF
factorization method with a Frobenius NMF initialization.</p>
<pre class="r"><code>log1p_fit_init_nmf &lt;- fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = 25,
  maxiter = 25,
  approx_range = c(0, 1.25),
  s = as.vector(sz),
  init_method = &quot;frob_nmf&quot;
)</code></pre>
</div>
<div id="analysis" class="section level2">
<h2>Analysis</h2>
</div>
<div id="heatmaps" class="section level1">
<h1>Heatmaps</h1>
<p>Below are the heatmaps of the different models:</p>
<pre class="r"><code>FF_log1p_greedy &lt;- log1p_greedy_fit$U
FF_log1p_greedy &lt;- scale(
  FF_log1p_greedy,
  center = FALSE, 
  scale = apply(FF_log1p_greedy, 2, max)
)
colnames(FF_log1p_greedy) &lt;- paste0(&quot;k&quot;, 1:ncol(FF_log1p_greedy))

FF_log1p_greedy_init &lt;- log1p_fit_init_greedy$U
FF_log1p_greedy_init &lt;- scale(
  FF_log1p_greedy_init, 
  center = FALSE,
  scale = apply(FF_log1p_greedy_init, 2, max)
)
colnames(FF_log1p_greedy_init) &lt;- paste0(&quot;k&quot;, 1:ncol(FF_log1p_greedy_init))

FF_log1p_nmf_init &lt;- log1p_fit_init_nmf$U
FF_log1p_nmf_init &lt;- scale(
  FF_log1p_nmf_init, 
  center = FALSE,
  scale = apply(FF_log1p_nmf_init, 2, max)
)
colnames(FF_log1p_nmf_init) &lt;- paste0(&quot;k&quot;, 1:ncol(FF_log1p_nmf_init))

cell.type &lt;- samples$celltype

# Downsample the number of cells and sort them using tSNE.
set.seed(8675309)
cell.idx &lt;- numeric(0)
cell.types &lt;- levels(cell.type)
for (i in 1:length(cell.types)) {
  which.idx &lt;- which(cell.type == cell.types[i])
  # Downsample common cell types.
  if (length(which.idx) &gt; 1250) {
    which.idx &lt;- sample(which.idx, 1250)
  }
  # Don&#39;t include rare cell types.
  if (length(which.idx) &gt; 20) {
    # Sort using tsne.
    tsne.res &lt;- Rtsne::Rtsne(
      FF_log1p_greedy[which.idx, ],
      dims = 1,
      pca = FALSE,
      normalize = FALSE,
      perplexity = min(100, floor((length(which.idx) - 1) / 3) - 1),
      theta = 0.1,
      max_iter = 1000,
      eta = 200,
      check_duplicates = FALSE
    )$Y[, 1]
    which.idx &lt;- which.idx[order(tsne.res)]
    cell.idx &lt;- c(cell.idx, which.idx)
  }
}

cell.type &lt;- cell.type[cell.idx]
cell.type &lt;- droplevels(cell.type)

FF_log1p_greedy &lt;- FF_log1p_greedy[cell.idx, ]
FF_log1p_greedy_init &lt;- FF_log1p_greedy_init[cell.idx, ]
FF_log1p_nmf_init &lt;- FF_log1p_nmf_init[cell.idx, ]

make.heatmap.tib &lt;- function(FF) {
  tib &lt;- as_tibble(scale(FF, center = FALSE, scale = apply(FF, 2, max))) %&gt;%
    mutate(Cell.type = cell.type) %&gt;%
    arrange(Cell.type) %&gt;%
    mutate(Cell.idx = row_number())
  
  tib &lt;- tib %&gt;%
    pivot_longer(
      -c(Cell.idx, Cell.type),
      names_to = &quot;Factor&quot;,
      values_to = &quot;Loading&quot;,
      values_drop_na = TRUE
    ) %&gt;%
    mutate(Factor = as.numeric(str_extract(Factor, &quot;[0-9]+&quot;)))
  
  return(tib)
}

log1p_greedy_tib &lt;- make.heatmap.tib(FF_log1p_greedy)
log1p_greedy_init_tib &lt;- make.heatmap.tib(FF_log1p_greedy_init)
log1p_nmf_init_tib &lt;- make.heatmap.tib(FF_log1p_nmf_init)

heatmap.tib &lt;- log1p_greedy_tib %&gt;% mutate(Method = &quot;log1p Poisson NMF Greedy Fit&quot;) %&gt;%
  bind_rows(log1p_greedy_init_tib %&gt;% mutate(Method = &quot;log1p Poisson NMF Greedy Init.&quot;)) %&gt;%
  bind_rows(log1p_nmf_init_tib %&gt;% mutate(Method = &quot;log1p Poisson NMF Frob NMF Init.&quot;)) %&gt;%
  mutate(Method = factor(Method, levels = c(&quot;log1p Poisson NMF Greedy Fit&quot;, &quot;log1p Poisson NMF Greedy Init.&quot;, &quot;log1p Poisson NMF Frob NMF Init.&quot;)))

tib &lt;- heatmap.tib %&gt;%
  group_by(Cell.type, Cell.idx) %&gt;%
  summarize()

cell_type_breaks &lt;- c(1, which(tib$Cell.type[-1] != tib$Cell.type[-nrow(tib)]))
label_pos &lt;- cell_type_breaks / 2 + c(cell_type_breaks[-1], nrow(tib)) / 2

library(ggplot2)

plt &lt;- ggplot(heatmap.tib, aes(x = Factor, y = -Cell.idx, fill = Loading)) +
  geom_tile() +
  scale_fill_gradient(low = &quot;white&quot;, high = &quot;firebrick&quot;) +
  labs(y = &quot;&quot;) +
  scale_y_continuous(breaks = -label_pos,
                     minor_breaks = NULL,
                     labels = levels(cell.type)) +
  scale_x_continuous(breaks = seq(0, 30, 5)) +
  theme_minimal() +
  geom_hline(yintercept = -cell_type_breaks, size = 0.1) +
  facet_wrap(~Method, ncol = 1, axes = &quot;all&quot;) +
  theme(legend.position = &quot;none&quot;, 
        strip.text = element_text(size = 16)) 

plt</code></pre>
<p><img src="figure/greedy_init.Rmd/unnamed-chunk-7-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/950f789f2b1c3eb293a8e98467f3d34cb511defa/docs/figure/greedy_init.Rmd/unnamed-chunk-7-1.png" target="_blank">950f789</a>
</td>
<td>
Eric Weine
</td>
<td>
2024-04-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The greedy fit looks generally as expected, except that the 24th
factor also looks like an intercept, which is strange. With greedy
initialization, the fit looks somewhere in between the NMF init and the
greedy init. The first factor stays mostly intact, but you can see that
some if it has been replaced by other factors.</p>
</div>
<div id="go-analysis" class="section level1">
<h1>GO Analysis</h1>
<p>First, we exclude genes that are very highly expressed across many
factors.</p>
<pre class="r"><code>rownames(log1p_greedy_fit$V) &lt;- genes_map_inv[rownames(log1p_greedy_fit$V)]
rownames(log1p_fit_init_greedy$V) &lt;- genes_map_inv[rownames(log1p_fit_init_greedy$V)]
rownames(log1p_fit_init_nmf$V) &lt;- genes_map_inv[rownames(log1p_fit_init_nmf$V)]

log1p_greedy_fit$V &lt;- log1p_greedy_fit$V[
  rownames(log1p_greedy_fit$V) != &quot;MALAT1&quot; &amp;
    !(rownames(log1p_greedy_fit$V) %in% rownames(log1p_greedy_fit$V)[grep(&quot;^RPS&quot;, rownames(log1p_greedy_fit$V))]) &amp;
    !(rownames(log1p_greedy_fit$V) %in% rownames(log1p_greedy_fit$V)[grep(&quot;^RPL&quot;, rownames(log1p_greedy_fit$V))]) &amp;
    !(rownames(log1p_greedy_fit$V) %in% rownames(log1p_greedy_fit$V)[grep(&quot;\\.&quot;, rownames(log1p_greedy_fit$V))]), 
]

log1p_fit_init_greedy$V &lt;- log1p_fit_init_greedy$V[
  rownames(log1p_fit_init_greedy$V) != &quot;MALAT1&quot; &amp;
    !(rownames(log1p_fit_init_greedy$V) %in% rownames(log1p_fit_init_greedy$V)[grep(&quot;^RPS&quot;, rownames(log1p_fit_init_greedy$V))]) &amp;
    !(rownames(log1p_fit_init_greedy$V) %in% rownames(log1p_fit_init_greedy$V)[grep(&quot;^RPL&quot;, rownames(log1p_fit_init_greedy$V))]) &amp;
    !(rownames(log1p_fit_init_greedy$V) %in% rownames(log1p_fit_init_greedy$V)[grep(&quot;\\.&quot;, rownames(log1p_fit_init_greedy$V))]), 
]

log1p_fit_init_nmf$V &lt;- log1p_fit_init_nmf$V[
  rownames(log1p_fit_init_nmf$V) != &quot;MALAT1&quot; &amp;
    !(rownames(log1p_fit_init_nmf$V) %in% rownames(log1p_fit_init_nmf$V)[grep(&quot;^RPS&quot;, rownames(log1p_fit_init_nmf$V))]) &amp;
    !(rownames(log1p_fit_init_nmf$V) %in% rownames(log1p_fit_init_nmf$V)[grep(&quot;^RPL&quot;, rownames(log1p_fit_init_nmf$V))]) &amp;
    !(rownames(log1p_fit_init_nmf$V) %in% rownames(log1p_fit_init_nmf$V)[grep(&quot;\\.&quot;, rownames(log1p_fit_init_nmf$V))]), 
]</code></pre>
<pre class="r"><code>library(clusterProfiler)
library(fgsea)
library(AnnotationDbi)
library(org.Hs.eg.db)

get_go_terms &lt;- function(V) {
  
  K &lt;- ncol(V)
  
  genes_vec &lt;- c()
  go_terms_vec &lt;- c()
  
  for (k in 1:K) {
    
    driving_genes &lt;- names(sort(V[,k], decreasing = TRUE))[1:20]
    driving_genes_ensembl &lt;- genes_map[driving_genes]
    
    
    go_result &lt;- enrichGO(gene = driving_genes_ensembl,
                          OrgDb = org.Hs.eg.db,
                          keyType = &quot;ENSEMBL&quot;,
                          ont = &quot;BP&quot;, 
                          pAdjustMethod = &quot;bonferroni&quot;,
                          pvalueCutoff = 0.01,
                          qvalueCutoff = 0.1)@result
    
    go_result &lt;- go_result %&gt;%
      dplyr::filter(p.adjust &lt; .01)
    
    go_terms &lt;- go_result$Description
    
    genes_vec &lt;- c(genes_vec, paste(driving_genes, collapse = &quot;, &quot;))
    go_terms_vec &lt;- c(go_terms_vec, paste(go_terms, collapse = &quot;, &quot;))
    
  }
  
  go_df &lt;- data.frame(
    driving_genes = genes_vec,
    go_terms = go_terms_vec,
    factor = 1:K
  )
  
  return(go_df)
  
}

gdf &lt;- get_go_terms(log1p_greedy_fit$V)
gdf2 &lt;- get_go_terms(log1p_fit_init_greedy$V)
gdf3 &lt;- get_go_terms(log1p_fit_init_nmf$V)</code></pre>
<p>Go Terms for log1p link Poisson NMF greedy fit</p>
<pre class="r"><code>knitr::kable(gdf)</code></pre>
<table>
<colgroup>
<col width="10%" />
<col width="88%" />
<col width="0%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">driving_genes</th>
<th align="left">go_terms</th>
<th align="right">factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">B2M, TMSB4X, TMSB10, ACTB, MT-CO1, HLA-B, TPT1, GNB2L1,
EEF1A1, FAU, FTL, HLA-A, PTMA, MT-CO3, UBA52, FTH1, NACA, MT-CO2, HLA-C,
PFN1</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, mitochondrial electron transport, cytochrome c to oxygen,
antigen processing and presentation of peptide antigen via MHC class I,
antigen processing and presentation of endogenous peptide antigen,
positive regulation of T cell mediated cytotoxicity, antigen processing
and presentation of endogenous antigen, regulation of T cell mediated
cytotoxicity, negative regulation of binding, T cell mediated
cytotoxicity, positive regulation of T cell mediated immunity, positive
regulation of leukocyte mediated cytotoxicity, positive regulation of
cell killing</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">GNLY, NKG7, S100A9, TYROBP, S100A8, CCL5, GZMB, FCER1G,
CST3, GZMA, LYZ, CST7, CLIC3, HCST, CYBA, S100A4, FGFBP2, KLRB1, CTSW,
LGALS1</td>
<td align="left">cell killing, killing of cells of another organism,
disruption of cell in another organism, disruption of anatomical
structure in another organism, defense response to bacterium</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">ZFAS1, AIF1, SPINK2, SNHG7, SERPINB1, HSP90AB1, PRSS57,
EIF3E, HINT1, GPX1, SOX4, KIAA0125, H2AFY, UQCRH, GSTP1, TALDO1, TXN,
UQCRB, CYTL1, SNHG8</td>
<td align="left"></td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">CD74, HLA-DRA, HLA-DRB1, HLA-DPB1, HLA-DPA1, CD79A,
CD79B, HLA-DRB5, HLA-DQA1, IGLL5, CD37, HLA-DQA2, MS4A1, HLA-DQB1,
TCL1A, HLA-DMA, HLA-DMB, LY86, LINC00926, NCF1</td>
<td align="left">antigen processing and presentation of exogenous
peptide antigen via MHC class II, antigen processing and presentation of
peptide antigen via MHC class II, antigen processing and presentation of
peptide or polysaccharide antigen via MHC class II, antigen processing
and presentation of exogenous peptide antigen, antigen processing and
presentation of exogenous antigen, MHC class II protein complex
assembly, peptide antigen assembly with MHC class II protein complex,
MHC protein complex assembly, peptide antigen assembly with MHC protein
complex, antigen processing and presentation of peptide antigen, antigen
processing and presentation, positive regulation of T cell activation,
positive regulation of leukocyte cell-cell adhesion, positive regulation
of lymphocyte activation, positive regulation of cell-cell adhesion,
lymphocyte proliferation, mononuclear cell proliferation, leukocyte
proliferation, antigen receptor-mediated signaling pathway, immune
response-activating cell surface receptor signaling pathway, dendritic
cell antigen processing and presentation, lymphocyte differentiation,
immune response-regulating cell surface receptor signaling pathway, T
cell activation involved in immune response, regulation of lymphocyte
proliferation, regulation of mononuclear cell proliferation, regulation
of leukocyte proliferation, positive regulation of lymphocyte
proliferation</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">IL32, JUNB, CD3E, LTB, CD3D, ACTB, IER2, AES, B2M,
GIMAP7, CORO1A, HLA-A, BTG1, CD27, EEF1A1, TMEM66, EVL, NOSIP, LCK,
UBC</td>
<td align="left">T cell differentiation, positive regulation of T cell
activation, positive regulation of leukocyte cell-cell adhesion,
positive regulation of lymphocyte activation, lymphocyte
differentiation, positive regulation of cell-cell adhesion</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">CD8B, CD3D, NOSIP, S100B, CD3E, RGS10, CD8A, CD7, CD27,
TMEM66, JUNB, CARS, EEF1A1, CCR7, CPA5, PRKCQ-AS1, TMSB10, GIMAP7, LEF1,
TCF7</td>
<td align="left">T cell differentiation, lymphocyte differentiation, T
cell receptor signaling pathway, thymic T cell selection, antigen
receptor-mediated signaling pathway, gamma-delta T cell activation, T
cell selection</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">CCL5, GZMK, DUSP1, ID2, PNRC1, CIRBP, MT-CYB, MT-ND2,
HOPX, MT-ND5, TXNIP, MT-CO3, TSC22D3, JUN, MT-CO2, MT-ND4, H3F3B,
MT-CO1, RBM39, CD69</td>
<td align="left">aerobic electron transport chain, ATP synthesis coupled
electron transport, mitochondrial ATP synthesis coupled electron
transport, respiratory electron transport chain, oxidative
phosphorylation, electron transport chain, aerobic respiration, cellular
respiration, proton transmembrane transport, energy derivation by
oxidation of organic compounds, mitochondrial electron transport,
cytochrome c to oxygen, mitochondrial electron transport, NADH to
ubiquinone</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">ADCK2, ZNF7, DGKE, RBM12, MBTPS1, POLDIP3, SDCCAG3,
ALKBH2, MYBBP1A, MXI1, NUP188, SEC22A, GLI4, GSTT1, RECQL5, THUMPD1,
RABL2A, TMEM41A, REV1, RSAD1</td>
<td align="left"></td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">TUBA1B, HMGB2, TUBB, STMN1, H2AFZ, PTMA, DUT,
HNRNPA2B1, PCNA, RAN, HMGB1, PPIA, RANBP1, GAPDH, HIST1H4C, SNRPB,
H2AFV, TYMS, YBX1, CHCHD2</td>
<td align="left">regulation of nuclease activity, positive regulation of
binding</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">PDCD4, BOD1, C15orf40, NLRP1, PDE7A, TSPYL1, TRMT61B,
PEX13, PTRH2, KIAA1328, ZNF559, AGER, MTERF, HSBP1L1, POM121, SMPD2,
JRKL, STPG1, USP32, POM121C</td>
<td align="left"></td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">BTG1, TXNIP, DDX5, CD7, GIMAP7, SELL, TMIGD2, CCR7,
FAIM3, C1orf162, STK17A, VAMP2, ANAPC16, ATP6V1G1, EIF4A2, TMEM66,
LITAF, ADD3, ZNF75A, LIMD2</td>
<td align="left"></td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">PRSS1, KIAA0125, HOPX, SPINK2, IGJ, LST1, PRSS3,
TSC22D1, CD99, ACTG1, ITM2C, DNTT, SNHG7, MZB1, ENO1, HOXA9, C9orf89,
ADA, KLF6, RNASEH2B</td>
<td align="left"></td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">S100A8, S100A9, AIF1, FCN1, LST1, FTL, LYZ, FTH1, TYMP,
GPX1, SAT1, PSAP, S100A6, TSPO, CFD, CST3, CD14, S100A11, CTSS, FOS</td>
<td align="left">sequestering of metal ion, maintenance of location in
cell</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td align="left">MT-CO1, MT-CO3, MT-CYB, MT-ND1, MT-ND4, MT-CO2, SPCS1,
XBP1, FKBP11, MT-ND2, C15orf61, SEC11C, METTL2B, WRAP73, TTC1, MIR142,
C12orf57, ESYT2, TINF2, SEPT2</td>
<td align="left">aerobic electron transport chain, ATP synthesis coupled
electron transport, mitochondrial ATP synthesis coupled electron
transport, respiratory electron transport chain, oxidative
phosphorylation, electron transport chain, aerobic respiration, cellular
respiration, energy derivation by oxidation of organic compounds, proton
transmembrane transport, mitochondrial electron transport, cytochrome c
to oxygen, mitochondrial electron transport, NADH to ubiquinone</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">GZMB, KLRB1, CLIC3, FGFBP2, PRF1, CST7, GZMA, CD7,
SPON2, PTGDS, CTSW, FCGR3A, PLAC8, CD247, GZMM, KLRF1, CCL4, IFITM2,
IFITM1, GZMH</td>
<td align="left">cell killing, killing of cells of another organism,
disruption of cell in another organism, disruption of anatomical
structure in another organism, leukocyte mediated cytotoxicity</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">MPHOSPH8, DYNLL2, MLLT10, DUS1L, GPATCH4, MYH9, STAT5B,
PRPF3, UNK, HIVEP2, PELP1, ARHGEF18, TRIM41, NUDT7, HNRNPUL2, FGD5-AS1,
STK24, VEGFB, FXYD7, FBXW4</td>
<td align="left"></td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">TPSAB1, LMO4, S100A6, APOC1, CD63, S100A4, MT-ATP6,
PRSS57, MT-ND1, MT-ND3, LINC00152, C4orf48, CNRIP1, ATP5E, POLR2L,
S100A8, TMEM258, HDC, SEC61B, EXD3</td>
<td align="left">proton motive force-driven mitochondrial ATP synthesis,
proton motive force-driven ATP synthesis, ATP biosynthetic process,
purine ribonucleoside triphosphate biosynthetic process, purine
nucleoside triphosphate biosynthetic process, ribonucleoside
triphosphate biosynthetic process, nucleoside triphosphate biosynthetic
process, ATP metabolic process, oxidative phosphorylation, purine
ribonucleoside triphosphate metabolic process, purine nucleoside
triphosphate metabolic process, ribonucleoside triphosphate metabolic
process, nucleoside triphosphate metabolic process</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">FAM98C, DHRS1, FAM78A, MED24, PDSS2, ALG13, COX10,
CPEB4, XPNPEP3, CCDC22, MUTYH, TAB1, DDX19A, ZFAND3, THAP4, PSME3,
COPS7A, FBXO25, DCP1A, ACTR3B</td>
<td align="left"></td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">CCL5, IL32, FOS, DUSP1, JUN, NFKBIA, GZMK, ZFP36,
S100A4, DUSP2, LYAR, CXCR4, IL7R, S100A10, KLF6, PPP1R15A, TSC22D3,
ZFP36L2, S100A6, ANXA1</td>
<td align="left">regulation of hemopoiesis, response to glucocorticoid,
cellular response to epidermal growth factor stimulus, integrated stress
response signaling, response to epidermal growth factor</td>
<td align="right">19</td>
</tr>
<tr class="even">
<td align="left">TIGIT, PPBP, PF4, FANK1, PMAIP1, IPPK, CNST, FOXP3,
IL2RA, ABTB1, ANKFY1, VPS18, HECTD3, NABP1, RIPK3, S1PR4, CD70, STAM,
RNF214, ENTPD1</td>
<td align="left">T cell homeostasis, lymphocyte homeostasis, leukocyte
homeostasis</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">ECHDC1, B4GALT7, UFL1, R3HCC1L, DDX42, RNF2, RFWD2,
HPS3, BLOC1S5, RCE1, SNX15, SLC30A7, LRRC23, CREB1, DERA, SPECC1L,
KLHL9, DHX58, RRP8, ALAS1</td>
<td align="left"></td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">EEF1A1, PABPC1, GLTSCR2, EEF2, EEF1D, EIF3L, LDHB,
TMEM66, ST13, GNB2L1, EIF3H, NOSIP, TPT1, HNRNPA1, CD3E, GPSM3, MAL,
LEF1, GYPC, NPM1</td>
<td align="left">translational elongation</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">ACTG1, ACTB, ARPC1B, PFN1, COTL1, CORO1A, MYL6, ARPC2,
CFL1, GAPDH, TMSB10, VIM, ARHGDIB, UCP2, CAPZB, LGALS1, CNN2, CLIC1,
PPP1CA, LSP1</td>
<td align="left">actin polymerization or depolymerization, regulation of
actin filament-based process, regulation of actin filament
polymerization, regulation of actin polymerization or depolymerization,
regulation of actin filament organization, regulation of actin filament
length, actin filament polymerization, regulation of actin cytoskeleton
organization, regulation of protein polymerization, regulation of
cellular component size, regulation of supramolecular fiber
organization, protein polymerization, regulation of protein-containing
complex assembly</td>
<td align="right">23</td>
</tr>
<tr class="even">
<td align="left">S100A4, LGALS1, CRIP1, S100A6, S100A11, LTB, S100A10,
CD52, TMSB10, EMP3, SH3BGRL3, GSTK1, PLP2, VIM, TRADD, IFITM1, TMSB4X,
TMA7, CORO1B, FXYD5</td>
<td align="left"></td>
<td align="right">24</td>
</tr>
<tr class="odd">
<td align="left">NOC2L, KLHL17, ISG15, C1orf159, TNFRSF18, TNFRSF4,
B3GALT6, ACAP3, CPSF3L, GLTPD1, MXRA8, CCNL2, MRPL20, ANKRD65, ATAD3B,
SSU72, C1orf233, MIB2, MMP23B, SLC35E2</td>
<td align="left"></td>
<td align="right">25</td>
</tr>
<tr class="even">
<td align="left">MAFF, UBR2, PPFIBP2, CCDC125, BAK1, DGKH, PLCL1, PQLC3,
NSG1, AMMECR1L, SIN3A, CAPG, EXOC2, COQ5, MDS2, BTAF1, TFE3, TMEM65,
ZNF681, PRTFDC1</td>
<td align="left"></td>
<td align="right">26</td>
</tr>
</tbody>
</table>
<p>Go Terms for log1p link Poisson NMF greedy initialization</p>
<pre class="r"><code>knitr::kable(gdf2)</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="92%" />
<col width="0%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">driving_genes</th>
<th align="left">go_terms</th>
<th align="right">factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">FAU, TPT1, GNB2L1, UBA52, B2M, EEF1A1, NACA, TMSB4X,
FTL, PTMA, EEF1B2, PFDN5, EEF1D, MT-CO1, COX4I1, TMSB10, GLTSCR2,
MT-CO3, EIF1, ACTB</td>
<td align="left">translational elongation, mitochondrial electron
transport, cytochrome c to oxygen, negative regulation of binding</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">GNLY, HLA-B, HLA-C, B2M, IFITM2, TMSB4X, HLA-A, HCST,
TMSB10, PFN1, NKG7, ACTB, ATP5E, MYL6, CYBA, CFL1, SH3BGRL3, TMA7,
PSME1, UBB</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, cell killing, antigen processing and presentation of
peptide antigen via MHC class I, antigen processing and presentation of
endogenous peptide antigen, positive regulation of T cell mediated
cytotoxicity, antigen processing and presentation of endogenous antigen,
antigen processing and presentation, regulation of T cell mediated
cytotoxicity, T cell mediated cytotoxicity, positive regulation of T
cell mediated immunity, positive regulation of leukocyte mediated
cytotoxicity, leukocyte mediated cytotoxicity, positive regulation of
cell killing</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">ZFAS1, NPM1, EIF3E, SERPINB1, HINT1, AIF1, LDHB, ACTG1,
GAPDH, MT-CO2, SNHG7, CD74, H3F3B, VIM, HSP90AB1, PRSS57, HLA-DRA,
MT-ND2, CLIC1, TAGLN2</td>
<td align="left">response to type II interferon, cellular response to
type II interferon</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">CD74, HLA-DRA, HLA-DRB1, LTB, HLA-DPB1, CD79A,
HLA-DPA1, CD52, CD79B, HLA-DRB5, CD37, IGLL5, HLA-DQA1, PTPRCAP, MS4A1,
HLA-DQA2, CYBA, TMSB4X, HLA-DQB1, TMSB10</td>
<td align="left">antigen processing and presentation of exogenous
peptide antigen via MHC class II, antigen processing and presentation of
peptide antigen via MHC class II, antigen processing and presentation of
peptide or polysaccharide antigen via MHC class II, antigen processing
and presentation of exogenous peptide antigen, antigen processing and
presentation of exogenous antigen, MHC class II protein complex
assembly, peptide antigen assembly with MHC class II protein complex,
MHC protein complex assembly, peptide antigen assembly with MHC protein
complex, antigen processing and presentation of peptide antigen, antigen
processing and presentation, positive regulation of T cell activation,
positive regulation of leukocyte cell-cell adhesion, positive regulation
of lymphocyte activation, positive regulation of cell-cell adhesion,
immune response-activating cell surface receptor signaling pathway,
antigen receptor-mediated signaling pathway, immune response-regulating
cell surface receptor signaling pathway, lymphocyte proliferation,
mononuclear cell proliferation, leukocyte proliferation, dendritic cell
antigen processing and presentation, lymphocyte differentiation</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">ACTB, ACTG1, PFN1, CORO1A, CFL1, ARPC1B, RAC2, CYBA,
SH3BGRL3, ARPC2, MYL12A, MYL6, PTPRCAP, COTL1, ARHGDIB, S100A4, CLIC1,
EMP3, B2M, GAPDH</td>
<td align="left">platelet aggregation, actin polymerization or
depolymerization, homotypic cell-cell adhesion, regulation of actin
filament organization, tissue homeostasis, anatomical structure
homeostasis, regulation of actin filament polymerization, platelet
activation, regulation of actin polymerization or depolymerization,
regulation of actin filament length</td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">TMSB10, LTB, CD52, CD8B, CD3D, TMSB4X, CD3E, LDHB,
HCST, NOSIP, HLA-C, JUNB, CD7, HLA-B, PTPRCAP, BTG1, IL32, NPM1, CORO1A,
S100B</td>
<td align="left"></td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">B2M, PTMA, FTL, HLA-B, FTH1, HLA-A, EIF1, HLA-C, UBC,
CIRBP, TXNIP, H3F3B, DDX5, TSC22D3, PNRC1, HMGB1, HLA-E, SRSF5, TMSB4X,
HNRNPDL</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
regulation of T cell mediated immunity, antigen processing and
presentation of endogenous peptide antigen via MHC class I, antigen
processing and presentation of peptide antigen via MHC class I, antigen
processing and presentation of endogenous peptide antigen, T cell
mediated immunity, positive regulation of T cell mediated cytotoxicity,
antigen processing and presentation of endogenous antigen, regulation of
T cell mediated cytotoxicity, T cell mediated cytotoxicity, positive
regulation of T cell mediated immunity, positive regulation of leukocyte
mediated cytotoxicity, regulation of adaptive immune response based on
somatic recombination of immune receptors built from immunoglobulin
superfamily domains, positive regulation of cell killing, regulation of
adaptive immune response, antigen processing and presentation of
endogenous peptide antigen via MHC class I via ER pathway,
TAP-independent, regulation of lymphocyte mediated immunity, antigen
processing and presentation of endogenous peptide antigen via MHC class
Ib, antigen processing and presentation of endogenous peptide antigen
via MHC class I via ER pathway, positive regulation of adaptive immune
response based on somatic recombination of immune receptors built from
immunoglobulin superfamily domains, positive regulation of adaptive
immune response, antigen processing and presentation of peptide antigen,
regulation of leukocyte mediated immunity, regulation of leukocyte
mediated cytotoxicity, regulation of cell killing, positive regulation
of lymphocyte mediated immunity, protection from natural killer cell
mediated cytotoxicity, adaptive immune response based on somatic
recombination of immune receptors built from immunoglobulin superfamily
domains, antigen processing and presentation, positive regulation of
leukocyte mediated immunity, leukocyte mediated cytotoxicity</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">IER2, JUNB, FTL, HLA-B, HLA-C, B2M, HLA-A, JUN, ZFP36,
H3F3B, UBC, S100A6, PTMA, FTH1, OAZ1, CYBA, SH3BGRL3, BTG2, PPDPF,
MT-ND4</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, antigen processing and presentation of peptide antigen via
MHC class I, antigen processing and presentation of endogenous peptide
antigen, positive regulation of T cell mediated cytotoxicity, antigen
processing and presentation of endogenous antigen, regulation of T cell
mediated cytotoxicity, regulation of adaptive immune response based on
somatic recombination of immune receptors built from immunoglobulin
superfamily domains, T cell mediated cytotoxicity, positive regulation
of T cell mediated immunity, positive regulation of leukocyte mediated
cytotoxicity, regulation of adaptive immune response, positive
regulation of cell killing</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">TUBA1B, PTMA, GAPDH, YBX1, STMN1, PFN1, HSP90AA1,
H2AFZ, CFL1, MT-CO2, ACTB, RAN, PPIA, HMGB1, TPI1, PSMA7, ENO1, SLC25A5,
RANBP1, CHCHD2</td>
<td align="left"></td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">MT-CO1, MT-CYB, MT-CO3, MT-CO2, MT-ND4, MT-ND2, MT-ND1,
MT-ATP6, MT-ND5, MT-ND3, BTG1, TMSB10, EEF1D, H3F3B, TMSB4X, COX4I1,
UBB, DDX5, SERF2, NEAT1</td>
<td align="left">oxidative phosphorylation, aerobic electron transport
chain, ATP synthesis coupled electron transport, mitochondrial ATP
synthesis coupled electron transport, aerobic respiration, respiratory
electron transport chain, cellular respiration, electron transport
chain, energy derivation by oxidation of organic compounds, proton
transmembrane transport, ATP biosynthetic process, purine ribonucleoside
triphosphate biosynthetic process, purine nucleoside triphosphate
biosynthetic process, ribonucleoside triphosphate biosynthetic process,
nucleoside triphosphate biosynthetic process, proton motive force-driven
mitochondrial ATP synthesis, ATP metabolic process, proton motive
force-driven ATP synthesis, purine ribonucleoside triphosphate metabolic
process, purine nucleoside triphosphate metabolic process,
ribonucleoside triphosphate metabolic process, nucleoside triphosphate
metabolic process, purine ribonucleotide biosynthetic process,
ribonucleotide biosynthetic process, ribose phosphate biosynthetic
process, mitochondrial electron transport, NADH to ubiquinone, purine
nucleotide biosynthetic process, purine-containing compound biosynthetic
process, mitochondrial electron transport, cytochrome c to oxygen,
nucleotide biosynthetic process, nucleoside phosphate biosynthetic
process, purine ribonucleotide metabolic process, ribonucleotide
metabolic process, ribose phosphate metabolic process, NADH
dehydrogenase complex assembly, mitochondrial respiratory chain complex
I assembly, mitochondrial respiratory chain complex assembly</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">TMSB4X, JUNB, BTG1, LTB, GIMAP7, CD3E, CD3D, TMSB10,
CD52, LDHB, CD27, LIMD2, CD7, TMEM66, NOSIP, CORO1A, ARHGDIB, PTPRCAP,
SELL, GYPC</td>
<td align="left"></td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">CD74, SPINK2, HLA-DRA, KIAA0125, AIF1, ZFAS1, LST1,
SNHG7, PRSS1, ACTG1, HOPX, LTB, GSTP1, CD52, HLA-DPB1, GPX1, ATP5E,
ACY3, ENO1, TMSB4X</td>
<td align="left">regulation of leukocyte proliferation</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">S100A9, S100A8, TYROBP, FTL, S100A4, TMSB4X, FTH1, LYZ,
AIF1, S100A6, CST3, HLA-DRB1, LST1, TMSB10, FCN1, SAT1, TSPO, CYBA,
PSAP, CD74</td>
<td align="left">maintenance of location in cell, sequestering of metal
ion, maintenance of location, positive regulation of interleukin-6
production</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td align="left">MT-CO2, MT-ND2, MT-CYB, MT-ATP6, MT-ND4, EIF3E, B2M,
SKP1, MT-ND5, SUB1, COX6C, PSMA2, MYL12A, C14orf2, TMSB4X, UQCRB, NACA,
SSBP1, BTF3, LDHB</td>
<td align="left">oxidative phosphorylation, aerobic electron transport
chain, ATP synthesis coupled electron transport, mitochondrial ATP
synthesis coupled electron transport, aerobic respiration, respiratory
electron transport chain, cellular respiration, electron transport
chain, energy derivation by oxidation of organic compounds, ATP
biosynthetic process, purine ribonucleoside triphosphate biosynthetic
process, purine nucleoside triphosphate biosynthetic process,
ribonucleoside triphosphate biosynthetic process, nucleoside
triphosphate biosynthetic process, ATP metabolic process, proton
transmembrane transport, purine ribonucleoside triphosphate metabolic
process, purine nucleoside triphosphate metabolic process,
ribonucleoside triphosphate metabolic process, proton motive
force-driven ATP synthesis, nucleoside triphosphate metabolic process,
purine ribonucleotide biosynthetic process, ribonucleotide biosynthetic
process, ribose phosphate biosynthetic process, purine nucleotide
biosynthetic process, purine-containing compound biosynthetic process,
nucleotide biosynthetic process, nucleoside phosphate biosynthetic
process, proton motive force-driven mitochondrial ATP synthesis, purine
ribonucleotide metabolic process, ribonucleotide metabolic process,
ribose phosphate metabolic process, mitochondrial electron transport,
NADH to ubiquinone</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">NKG7, TYROBP, GZMB, FCER1G, GZMA, CCL5, CST7, CLIC3,
KLRB1, FGFBP2, HOPX, PRF1, PLAC8, PTPRCAP, CD247, SRGN, GZMM, S100A4,
CD7, GZMH</td>
<td align="left">killing of cells of another organism, disruption of
cell in another organism, cell killing, disruption of anatomical
structure in another organism, leukocyte mediated cytotoxicity</td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">DUSP1, JUN, FOS, JUNB, NFKBIA, ZFP36, IER2, H3F3B,
TSC22D3, BTG1, PPP1R15A, CXCR4, CD69, EIF1, UBC, PNRC1, KLF6, ZFP36L2,
BTG2, DUSP2</td>
<td align="left">integrated stress response signaling, cellular response
to epidermal growth factor stimulus, response to epidermal growth
factor, nuclear-transcribed mRNA catabolic process</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">S100A6, S100A4, ATP5E, TMSB10, COX7C, ATP5L, ZFAS1,
MT-ND3, SERF2, TOMM7, SEC61B, MT-ATP6, UQCRB, UQCRH, LMO4, COX6B1, FAU,
LGALS1, ACTG1, OST4</td>
<td align="left">oxidative phosphorylation, aerobic respiration,
cellular respiration, energy derivation by oxidation of organic
compounds, aerobic electron transport chain, ATP synthesis coupled
electron transport, mitochondrial ATP synthesis coupled electron
transport, respiratory electron transport chain, proton transmembrane
transport, electron transport chain, proton motive force-driven
mitochondrial ATP synthesis, proton motive force-driven ATP synthesis,
ATP biosynthetic process, purine ribonucleoside triphosphate
biosynthetic process, purine nucleoside triphosphate biosynthetic
process, ribonucleoside triphosphate biosynthetic process, nucleoside
triphosphate biosynthetic process, ATP metabolic process, purine
ribonucleoside triphosphate metabolic process, purine nucleoside
triphosphate metabolic process, ribonucleoside triphosphate metabolic
process, nucleoside triphosphate metabolic process</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">FTH1, FTL, CD37, TMSB10, H3F3B, EIF1, OAZ1, UBB,
MT-CO1, COX4I1, GPSM3, UBXN1, DRAP1, LIMD2, AES, SH3BGRL3, GPX1, FXYD5,
EMP3, ERP29</td>
<td align="left"></td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">CCL5, IL32, GZMK, NKG7, GZMA, S100A4, LYAR, CD52, CST7,
CD3D, PTPRCAP, CTSW, S100A6, CD3E, DUSP2, CD99, HCST, BTG1, GZMM,
ID2</td>
<td align="left"></td>
<td align="right">19</td>
</tr>
<tr class="even">
<td align="left">IL32, TMSB4X, CD52, HLA-A, UCP2, S100A4, MT-CO1, B2M,
TMSB10, ARHGDIB, CD3D, GMFG, S100A10, CD27, CD74, ANXA2, PTPRCAP, GAPDH,
ISG20, LIMD2</td>
<td align="left"></td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">BTG1, DDX5, TXNIP, ACTB, HNRNPA2B1, EIF4A2, CNBP,
MYL12A, CD53, SRSF3, SRSF5, CCNI, RAN, ITM2B, HSPA8, TMEM66, B2M,
PTGES3, CD164, TMBIM6</td>
<td align="left">regulation of RNA splicing, primary miRNA processing,
regulation of mRNA splicing, via spliceosome, regulation of mRNA
processing, regulation of protein binding, miRNA processing, RNA
splicing, via transesterification reactions with bulged adenosine as
nucleophile, mRNA splicing, via spliceosome</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">EEF1A1, LTB, EEF2, HLA-C, HLA-B, VIM, PABPC1, NPM1,
HNRNPA1, GNB2L1, SLC25A6, GLTSCR2, LDHB, B2M, TPT1, EEF1B2, EIF3L,
HLA-A, EEF1D, DDX5</td>
<td align="left">translational elongation, antigen processing and
presentation of peptide antigen via MHC class Ib, antigen processing and
presentation via MHC class Ib, antigen processing and presentation of
endogenous peptide antigen via MHC class I, antigen processing and
presentation of peptide antigen via MHC class I, antigen processing and
presentation of endogenous peptide antigen, positive regulation of T
cell mediated cytotoxicity, antigen processing and presentation of
endogenous antigen, regulation of T cell mediated cytotoxicity, T cell
mediated cytotoxicity, positive regulation of T cell mediated immunity,
positive regulation of leukocyte mediated cytotoxicity, positive
regulation of cell killing, positive regulation of translation</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">CD74, HLA-DRA, HLA-DPB1, CST3, HLA-DPA1, HLA-DRB1, LYZ,
HLA-DQA1, HLA-DRB5, ACTB, VIM, S100A10, TMSB4X, HLA-DQB1, HLA-DQA2,
LGALS1, ACTG1, FTH1, CYBA, HLA-DMA</td>
<td align="left">antigen processing and presentation of exogenous
peptide antigen via MHC class II, antigen processing and presentation of
peptide antigen via MHC class II, antigen processing and presentation of
peptide or polysaccharide antigen via MHC class II, antigen processing
and presentation of exogenous peptide antigen, antigen processing and
presentation of exogenous antigen, MHC class II protein complex
assembly, peptide antigen assembly with MHC class II protein complex,
positive regulation of T cell activation, MHC protein complex assembly,
peptide antigen assembly with MHC protein complex, positive regulation
of leukocyte cell-cell adhesion, positive regulation of lymphocyte
activation, antigen processing and presentation of peptide antigen,
positive regulation of cell-cell adhesion, antigen processing and
presentation, positive regulation of viral entry into host cell,
positive regulation by symbiont of entry into host, dendritic cell
antigen processing and presentation, positive regulation of viral life
cycle, positive regulation of T cell differentiation</td>
<td align="right">23</td>
</tr>
<tr class="even">
<td align="left">LTB, S100A4, CD52, IL32, TMSB4X, TMSB10, S100A10, VIM,
B2M, S100A6, ANXA1, CRIP1, LGALS1, CD3D, SH3BGRL3, HLA-C, HLA-B, IL7R,
S100A11, NOSIP</td>
<td align="left">regulation of T cell mediated cytotoxicity, regulation
of adaptive immune response based on somatic recombination of immune
receptors built from immunoglobulin superfamily domains, T cell mediated
cytotoxicity, regulation of adaptive immune response</td>
<td align="right">24</td>
</tr>
<tr class="odd">
<td align="left">EI24, USF2, DDRGK1, FNBP1, NONO, CNOT2, NDUFA5,
NUDT16L1, DNMT1, RNF166, NASP, STAU1, UPF3A, NDUFAF2, COA4, KIF5B,
RBBP6, PNN, PQLC3, BRI3</td>
<td align="left"></td>
<td align="right">25</td>
</tr>
<tr class="even">
<td align="left">MAFF, UBR2, PPFIBP2, CCDC125, BAK1, DGKH, PLCL1, PQLC3,
NSG1, AMMECR1L, SIN3A, CAPG, EXOC2, COQ5, MDS2, BTAF1, TFE3, TMEM65,
ZNF681, PRTFDC1</td>
<td align="left"></td>
<td align="right">26</td>
</tr>
</tbody>
</table>
<p>Go Terms for log1p link Poisson NMF Frob. NMF initialization</p>
<pre class="r"><code>knitr::kable(gdf3)</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="92%" />
<col width="0%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">driving_genes</th>
<th align="left">go_terms</th>
<th align="right">factor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">LTB, GIMAP7, CD3E, TMSB4X, LDHB, B2M, BTG1, DDX5, CD3D,
CD27, TMEM66, NOSIP, TPT1, LIMD2, ARHGDIB, GNB2L1, PABPC1, EEF1A1, CD52,
SELL</td>
<td align="left"></td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">PTMA, TUBA1B, ACTB, HSP90AA1, YBX1, PFN1, CFL1, TMSB10,
TMSB4X, STMN1, H2AFZ, CALM2, CHCHD2, HMGB1, PPIA, SERF2, RAN, GAPDH,
PSMA7, GSTP1</td>
<td align="left">negative regulation of protein-containing complex
assembly, negative regulation of protein polymerization, regulation of
protein polymerization, regulation of protein-containing complex
assembly, regulation of binding, protein polymerization</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">EEF1B2, EEF1A1, TPT1, HNRNPA1, BTG1, NPM1, FAU, FTL,
PFDN5, GNB2L1, UBA52, NACA, TOMM7, BTF3, GLTSCR2, EIF1, HLA-B, SLC25A6,
EEF1D, PTMA</td>
<td align="left">translational elongation</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">IL32, S100A4, B2M, HLA-A, ACTB, HLA-B, TMSB4X, CD3D,
CD52, LGALS1, PTPRCAP, SH3BGRL3, HLA-C, S100A10, S100A6, PFN1, CD3E,
ARHGDIB, CORO1A, TMSB10</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
positive regulation of T cell activation, antigen processing and
presentation of endogenous peptide antigen via MHC class I, positive
regulation of leukocyte cell-cell adhesion, antigen processing and
presentation of peptide antigen via MHC class I, antigen processing and
presentation of endogenous peptide antigen, positive regulation of T
cell mediated cytotoxicity, antigen processing and presentation of
endogenous antigen, positive regulation of lymphocyte activation,
positive regulation of cell-cell adhesion, regulation of T cell mediated
cytotoxicity, regulation of actin filament organization, T cell mediated
cytotoxicity, positive regulation of T cell mediated immunity, positive
regulation of leukocyte mediated cytotoxicity, regulation of actin
filament polymerization, leukocyte mediated cytotoxicity, positive
regulation of cell killing, regulation of actin polymerization or
depolymerization</td>
<td align="right">4</td>
</tr>
<tr class="odd">
<td align="left">GNB2L1, NPM1, SERPINB1, FTL, NACA, GAPDH, MT-CO1, BTF3,
HNRNPA1, ZFAS1, FTH1, SNHG7, LDHB, AIF1, HSP90AB1, PRSS57, EIF3E,
EEF1B2, ATP5G2, TPT1</td>
<td align="left"></td>
<td align="right">5</td>
</tr>
<tr class="even">
<td align="left">GNLY, NKG7, B2M, TYROBP, GZMB, TMSB4X, FCER1G, HLA-A,
HLA-B, GZMA, CST7, CLIC3, HCST, HLA-C, IFITM2, KLRB1, CCL5, CTSW, ACTB,
FGFBP2</td>
<td align="left">cell killing, leukocyte mediated cytotoxicity, antigen
processing and presentation of peptide antigen via MHC class I,
regulation of leukocyte mediated cytotoxicity, regulation of cell
killing, positive regulation of leukocyte mediated cytotoxicity,
positive regulation of cell killing, antigen processing and presentation
of peptide antigen via MHC class Ib, antigen processing and presentation
via MHC class Ib, antigen processing and presentation of endogenous
peptide antigen via MHC class I, natural killer cell mediated
cytotoxicity, antigen processing and presentation of peptide antigen,
regulation of leukocyte mediated immunity, natural killer cell mediated
immunity, antigen processing and presentation of endogenous peptide
antigen, positive regulation of T cell mediated cytotoxicity, antigen
processing and presentation of endogenous antigen, antigen processing
and presentation, positive regulation of leukocyte mediated immunity,
killing of cells of another organism, disruption of cell in another
organism, regulation of T cell mediated cytotoxicity, negative
regulation of binding, T cell mediated cytotoxicity, disruption of
anatomical structure in another organism, positive regulation of T cell
mediated immunity, regulation of lymphocyte mediated immunity</td>
<td align="right">6</td>
</tr>
<tr class="odd">
<td align="left">CD74, HLA-DRA, HLA-DRB1, HLA-DPB1, HLA-DPA1, LTB,
CD79A, HLA-DRB5, CD79B, CD37, HLA-DQA1, CD52, IGLL5, HLA-DQA2, CYBA,
TMSB4X, FAU, PTPRCAP, MS4A1, HLA-DQB1</td>
<td align="left">antigen processing and presentation of exogenous
peptide antigen via MHC class II, antigen processing and presentation of
peptide antigen via MHC class II, antigen processing and presentation of
peptide or polysaccharide antigen via MHC class II, antigen processing
and presentation of exogenous peptide antigen, antigen processing and
presentation of exogenous antigen, MHC class II protein complex
assembly, peptide antigen assembly with MHC class II protein complex,
MHC protein complex assembly, peptide antigen assembly with MHC protein
complex, antigen processing and presentation of peptide antigen, antigen
processing and presentation, positive regulation of T cell activation,
positive regulation of leukocyte cell-cell adhesion, positive regulation
of lymphocyte activation, positive regulation of cell-cell adhesion,
immune response-activating cell surface receptor signaling pathway,
antigen receptor-mediated signaling pathway, immune response-regulating
cell surface receptor signaling pathway, lymphocyte proliferation,
mononuclear cell proliferation, leukocyte proliferation, dendritic cell
antigen processing and presentation, lymphocyte differentiation</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">ACTG1, MT-CO1, GAPDH, PFN1, SLC25A6, ACTB, TMSB10,
MT-CO3, ARHGDIB, TMSB4X, ARPC1B, CFL1, UCP2, GNB2L1, ARPC2, TAGLN2,
BTF3, MYL6, VIM, H3F3B</td>
<td align="left">actin polymerization or depolymerization, regulation of
protein-containing complex assembly, regulation of actin filament
organization, regulation of actin filament polymerization</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">CD74, SPINK2, HLA-DRA, ZFAS1, TPT1, B2M, AIF1, TMSB4X,
GNB2L1, KIAA0125, FAU, NACA, LST1, SNHG7, ACTG1, HOPX, GSTP1, HLA-DPB1,
PRSS1, H3F3B</td>
<td align="left">antigen processing and presentation of exogenous
peptide antigen via MHC class II, antigen processing and presentation of
peptide antigen via MHC class II, antigen processing and presentation of
peptide or polysaccharide antigen via MHC class II, antigen processing
and presentation of exogenous peptide antigen, antigen processing and
presentation of exogenous antigen, regulation of leukocyte
proliferation</td>
<td align="right">9</td>
</tr>
<tr class="even">
<td align="left">TMSB10, VIM, FAU, TPT1, UBA52, ATP5E, S100A6, COX7C,
ATP5L, ZFAS1, COMMD6, PTMA, S100A4, TMSB4X, TOMM7, EEF1B2, MT-CO3,
SERF2, NACA, HINT1</td>
<td align="left">oxidative phosphorylation, proton transmembrane
transport</td>
<td align="right">10</td>
</tr>
<tr class="odd">
<td align="left">VIM, B2M, TAGLN2, HLA-B, HLA-C, EEF1A1, GAPDH, S100A10,
ANXA1, PABPC1, UBC, EEF2, GNB2L1, HLA-A, FTL, FXYD5, YBX1, MT-CYB,
MT-ND4, SH3BGRL3</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, antigen processing and presentation of peptide antigen via
MHC class I, antigen processing and presentation of endogenous peptide
antigen, positive regulation of T cell mediated cytotoxicity, antigen
processing and presentation of endogenous antigen, regulation of T cell
mediated cytotoxicity, regulation of adaptive immune response based on
somatic recombination of immune receptors built from immunoglobulin
superfamily domains, T cell mediated cytotoxicity, positive regulation
of T cell mediated immunity, positive regulation of leukocyte mediated
cytotoxicity, regulation of adaptive immune response, positive
regulation of cell killing, positive regulation of translation</td>
<td align="right">11</td>
</tr>
<tr class="even">
<td align="left">CD8B, TMSB10, CD3D, IL32, TMSB4X, CD3E, HCST, CD52,
B2M, JUNB, CD7, NOSIP, S100B, ARHGDIB, LDHB, LTB, CTSW, CD27, HLA-B,
HLA-A</td>
<td align="left">T cell differentiation</td>
<td align="right">12</td>
</tr>
<tr class="odd">
<td align="left">ACTB, ACTG1, PFN1, CFL1, CORO1A, TMSB10, TMSB4X, CYBA,
MYL6, RAC2, IFITM2, ARPC1B, ARPC2, FTH1, B2M, EEF1D, EEF1A1, PTPRCAP,
SH3BGRL3, CD7</td>
<td align="left">actin polymerization or depolymerization, regulation of
actin filament polymerization, regulation of actin filament
organization, regulation of actin polymerization or depolymerization,
regulation of actin filament length, actin filament polymerization,
regulation of actin cytoskeleton organization, regulation of actin
filament-based process, regulation of protein polymerization, regulation
of supramolecular fiber organization, maintenance of location in cell,
tissue homeostasis, anatomical structure homeostasis, protein
polymerization</td>
<td align="right">13</td>
</tr>
<tr class="even">
<td align="left">EEF1A1, GNB2L1, EEF1D, TMSB4X, GLTSCR2, TPT1, EEF2,
NACA, UBA52, FAU, COX4I1, FTL, EEF1B2, TMSB10, SLC25A6, B2M, PABPC1,
FTH1, FXYD5, PFN1</td>
<td align="left">translational elongation, negative regulation of
protein-containing complex assembly</td>
<td align="right">14</td>
</tr>
<tr class="odd">
<td align="left">LTB, JUNB, NOSIP, IL7R, IL32, TPT1, EEF1A1, S100A4,
TMSB10, TIMP1, EEF1D, LDHB, TMSB4X, UBA52, FOS, IFITM1, IFITM2, B2M,
CD3E, H3F3B</td>
<td align="left"></td>
<td align="right">15</td>
</tr>
<tr class="even">
<td align="left">FTL, TMSB10, FTH1, B2M, EIF1, TMSB4X, EEF1D, PTMA, UBC,
HLA-B, CIRBP, MT-CO1, HLA-C, OAZ1, HLA-A, SH3BGRL3, COX4I1, TXNIP,
PFDN5, UBB</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, antigen processing and presentation of peptide antigen via
MHC class I, antigen processing and presentation of endogenous peptide
antigen, positive regulation of T cell mediated cytotoxicity, antigen
processing and presentation of endogenous antigen, regulation of T cell
mediated cytotoxicity, T cell mediated cytotoxicity, positive regulation
of T cell mediated immunity, positive regulation of leukocyte mediated
cytotoxicity, positive regulation of cell killing, regulation of T cell
mediated immunity</td>
<td align="right">16</td>
</tr>
<tr class="odd">
<td align="left">JUNB, H3F3B, CXCR4, FTH1, UBC, TSC22D3, B2M, BTG2,
EIF1, TMSB4X, BTG1, DDX5, DUSP2, H1FX, PTMA, ZFP36L2, ZFP36, TXNIP,
HLA-B, DDIT4</td>
<td align="left">nuclear-transcribed mRNA catabolic process, cellular
response to glucocorticoid stimulus</td>
<td align="right">17</td>
</tr>
<tr class="even">
<td align="left">CD52, TMSB4X, TMSB10, S100A4, FAU, TOMM7, TMA7, SERF2,
S100A6, CRIP1, B2M, ATP5E, UBA52, SH3BGRL3, GMFG, EMP3, UQCR11, FTL,
PFDN5, ATP5G2</td>
<td align="left">negative regulation of supramolecular fiber
organization</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td align="left">HLA-B, HLA-C, B2M, HLA-A, LTB, TMSB4X, EEF1A1, TPT1,
PSME1, FAU, EEF1B2, HLA-E, UBA52, ACTB, PTMA, PSMB9, GAPDH, ATP5E, ENO1,
HINT1</td>
<td align="left">antigen processing and presentation of peptide antigen
via MHC class Ib, antigen processing and presentation via MHC class Ib,
antigen processing and presentation of endogenous peptide antigen via
MHC class I, antigen processing and presentation of peptide antigen via
MHC class I, antigen processing and presentation of endogenous peptide
antigen, positive regulation of T cell mediated cytotoxicity, antigen
processing and presentation of endogenous antigen, regulation of T cell
mediated cytotoxicity, T cell mediated cytotoxicity, antigen processing
and presentation, positive regulation of T cell mediated immunity,
positive regulation of leukocyte mediated cytotoxicity, positive
regulation of cell killing, antigen processing and presentation of
endogenous peptide antigen via MHC class I via ER pathway,
TAP-independent, antigen processing and presentation of endogenous
peptide antigen via MHC class Ib, antigen processing and presentation of
endogenous peptide antigen via MHC class I via ER pathway, regulation of
T cell mediated immunity, positive regulation of adaptive immune
response based on somatic recombination of immune receptors built from
immunoglobulin superfamily domains, positive regulation of adaptive
immune response, antigen processing and presentation of peptide antigen,
T cell mediated immunity, cell killing, regulation of leukocyte mediated
cytotoxicity, regulation of cell killing, protection from natural killer
cell mediated cytotoxicity, positive regulation of lymphocyte mediated
immunity, positive regulation of leukocyte mediated immunity, regulation
of adaptive immune response based on somatic recombination of immune
receptors built from immunoglobulin superfamily domains, regulation of
adaptive immune response, leukocyte mediated cytotoxicity, antigen
processing and presentation of exogenous antigen, regulation of
lymphocyte mediated immunity, antimicrobial humoral response</td>
<td align="right">19</td>
</tr>
<tr class="even">
<td align="left">S100A9, S100A8, CST3, LYZ, FTL, TYROBP, HLA-DRB1, FTH1,
S100A4, CD74, HLA-DRA, TMSB4X, AIF1, S100A6, LGALS1, HLA-DPB1, CYBA,
LST1, FCER1G, HLA-DRB5</td>
<td align="left">sequestering of metal ion, antigen processing and
presentation of exogenous peptide antigen via MHC class II, antigen
processing and presentation of peptide antigen via MHC class II, antigen
processing and presentation of peptide or polysaccharide antigen via MHC
class II, antigen processing and presentation of exogenous peptide
antigen, antigen processing and presentation of exogenous antigen,
positive regulation of lymphocyte activation, positive regulation of T
cell activation, antigen processing and presentation of peptide antigen,
positive regulation of leukocyte cell-cell adhesion, maintenance of
location in cell, antigen processing and presentation, positive
regulation of cell-cell adhesion, regulation of lymphocyte
proliferation, regulation of mononuclear cell proliferation, leukocyte
activation involved in immune response, cell activation involved in
immune response, positive regulation of viral entry into host cell,
positive regulation by symbiont of entry into host, regulation of
leukocyte proliferation, maintenance of location, cell chemotaxis,
lymphocyte proliferation, mononuclear cell proliferation, MHC class II
protein complex assembly, peptide antigen assembly with MHC class II
protein complex, leukocyte proliferation, MHC protein complex assembly,
peptide antigen assembly with MHC protein complex, dendritic cell
antigen processing and presentation, positive regulation of viral life
cycle, lymphocyte activation involved in immune response, neutrophil
chemotaxis, leukocyte chemotaxis, myeloid leukocyte migration, positive
regulation of interleukin-6 production, T cell activation involved in
immune response, granulocyte chemotaxis, neutrophil migration</td>
<td align="right">20</td>
</tr>
<tr class="odd">
<td align="left">DUSP1, FOS, JUN, NFKBIA, PPP1R15A, TSC22D3, ZFP36,
KLF6, TUBA1A, CD69, BTG1, PNRC1, EIF1, UBC, TPT1, PABPC1, TMSB4X, YBX1,
ZFP36L2, ARL4A</td>
<td align="left">regulation of nuclear-transcribed mRNA catabolic
process, deadenylation-dependent decay, nuclear-transcribed mRNA
catabolic process, response to tumor necrosis factor, positive
regulation of nuclear-transcribed mRNA catabolic process,
deadenylation-dependent decay, nuclear-transcribed mRNA catabolic
process, deadenylation-dependent decay, negative regulation of DNA
binding, cellular response to tumor necrosis factor, cellular response
to epidermal growth factor stimulus, integrated stress response
signaling, response to epidermal growth factor</td>
<td align="right">21</td>
</tr>
<tr class="even">
<td align="left">MT-CO2, MT-ND2, MT-CYB, MT-ATP6, MT-ND4, B2M, EIF3E,
SKP1, MT-ND5, MYL12A, TMSB4X, BTF3, SUB1, COX6C, PSMA2, NACA, C14orf2,
NPM1, ACTB, SSBP1</td>
<td align="left">oxidative phosphorylation, aerobic electron transport
chain, aerobic respiration, ATP synthesis coupled electron transport,
mitochondrial ATP synthesis coupled electron transport, ATP biosynthetic
process, purine ribonucleoside triphosphate biosynthetic process, purine
nucleoside triphosphate biosynthetic process, ribonucleoside
triphosphate biosynthetic process, respiratory electron transport chain,
nucleoside triphosphate biosynthetic process, cellular respiration, ATP
metabolic process, proton transmembrane transport, purine ribonucleoside
triphosphate metabolic process, purine nucleoside triphosphate metabolic
process, ribonucleoside triphosphate metabolic process, electron
transport chain, proton motive force-driven ATP synthesis, energy
derivation by oxidation of organic compounds, nucleoside triphosphate
metabolic process, purine ribonucleotide biosynthetic process,
ribonucleotide biosynthetic process, ribose phosphate biosynthetic
process, purine nucleotide biosynthetic process, purine-containing
compound biosynthetic process, nucleotide biosynthetic process,
nucleoside phosphate biosynthetic process, proton motive force-driven
mitochondrial ATP synthesis, purine ribonucleotide metabolic process,
ribonucleotide metabolic process, ribose phosphate metabolic
process</td>
<td align="right">22</td>
</tr>
<tr class="odd">
<td align="left">IER2, JUN, FTL, ZFP36, JUNB, MT-CO1, BTG2, DUSP2,
GNB2L1, HLA-B, EEF1D, MT-CO3, PTMA, NFKBIA, UBA52, FAU, HLA-C, MT-ND1,
OAZ1, TMSB10</td>
<td align="left"></td>
<td align="right">23</td>
</tr>
<tr class="even">
<td align="left">MT-CYB, MT-CO1, MT-CO2, MT-CO3, MT-ND4, MT-ND2, MT-ND1,
MT-ATP6, MT-ND5, MT-ND3, FTL, COX4I1, ATP5D, UQCR11, CYBA, HLA-B, SERF2,
UBB, MZT2B, EEF1D</td>
<td align="left">oxidative phosphorylation, aerobic respiration, aerobic
electron transport chain, cellular respiration, ATP synthesis coupled
electron transport, mitochondrial ATP synthesis coupled electron
transport, electron transport chain, respiratory electron transport
chain, energy derivation by oxidation of organic compounds, proton
transmembrane transport, proton motive force-driven mitochondrial ATP
synthesis, proton motive force-driven ATP synthesis, ATP biosynthetic
process, purine ribonucleoside triphosphate biosynthetic process, purine
nucleoside triphosphate biosynthetic process, ribonucleoside
triphosphate biosynthetic process, nucleoside triphosphate biosynthetic
process, ATP metabolic process, purine ribonucleoside triphosphate
metabolic process, purine nucleoside triphosphate metabolic process,
ribonucleoside triphosphate metabolic process, nucleoside triphosphate
metabolic process, purine ribonucleotide biosynthetic process,
ribonucleotide biosynthetic process, ribose phosphate biosynthetic
process, mitochondrial electron transport, NADH to ubiquinone, purine
nucleotide biosynthetic process, purine-containing compound biosynthetic
process, mitochondrial electron transport, cytochrome c to oxygen,
nucleotide biosynthetic process, nucleoside phosphate biosynthetic
process, purine ribonucleotide metabolic process, ribonucleotide
metabolic process, ribose phosphate metabolic process, NADH
dehydrogenase complex assembly, mitochondrial respiratory chain complex
I assembly, mitochondrial respiratory chain complex assembly</td>
<td align="right">24</td>
</tr>
<tr class="odd">
<td align="left">CCL5, GNLY, NKG7, GZMK, IL32, GZMA, LYAR, B2M, HLA-B,
S100A4, CST7, HCST, HLA-C, CTSW, TMSB4X, HLA-A, CD3D, CD52, DUSP2,
ID2</td>
<td align="left">cell killing, antigen processing and presentation of
peptide antigen via MHC class Ib, antigen processing and presentation
via MHC class Ib, antigen processing and presentation of endogenous
peptide antigen via MHC class I, antigen processing and presentation of
peptide antigen via MHC class I, antigen processing and presentation of
endogenous peptide antigen, positive regulation of T cell mediated
cytotoxicity, antigen processing and presentation of endogenous antigen,
regulation of T cell mediated cytotoxicity, T cell mediated
cytotoxicity, positive regulation of T cell mediated immunity, positive
regulation of leukocyte mediated cytotoxicity, leukocyte mediated
cytotoxicity, positive regulation of cell killing, regulation of T cell
mediated immunity</td>
<td align="right">25</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>FALSE R version 4.3.2 (2023-10-31)
FALSE Platform: aarch64-apple-darwin20 (64-bit)
FALSE Running under: macOS Ventura 13.5
FALSE 
FALSE Matrix products: default
FALSE BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
FALSE LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
FALSE 
FALSE locale:
FALSE [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
FALSE 
FALSE time zone: America/New_York
FALSE tzcode source: internal
FALSE 
FALSE attached base packages:
FALSE [1] stats4    stats     graphics  grDevices utils     datasets  methods  
FALSE [8] base     
FALSE 
FALSE other attached packages:
FALSE  [1] org.Hs.eg.db_3.18.0    AnnotationDbi_1.64.1   IRanges_2.36.0        
FALSE  [4] S4Vectors_0.40.2       Biobase_2.62.0         BiocGenerics_0.48.1   
FALSE  [7] fgsea_1.28.0           clusterProfiler_4.10.0 ggplot2_3.5.0         
FALSE [10] stringr_1.5.1          purrr_1.0.2            tidyr_1.3.1           
FALSE [13] dplyr_1.1.4           
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] RColorBrewer_1.1-3            rstudioapi_0.15.0            
FALSE   [3] jsonlite_1.8.8                magrittr_2.0.3               
FALSE   [5] farver_2.1.1                  rmarkdown_2.26               
FALSE   [7] fs_1.6.3                      zlibbioc_1.48.2              
FALSE   [9] vctrs_0.6.5                   memoise_2.0.1                
FALSE  [11] RCurl_1.98-1.14               ggtree_3.10.1                
FALSE  [13] htmltools_0.5.7               curl_5.2.1                   
FALSE  [15] AnnotationHub_3.10.0          gridGraphics_0.5-1           
FALSE  [17] sass_0.4.8                    bslib_0.6.1                  
FALSE  [19] plyr_1.8.9                    cachem_1.0.8                 
FALSE  [21] whisker_0.4.1                 igraph_2.0.2                 
FALSE  [23] mime_0.12                     lifecycle_1.0.4              
FALSE  [25] pkgconfig_2.0.3               Matrix_1.6-5                 
FALSE  [27] R6_2.5.1                      fastmap_1.1.1                
FALSE  [29] gson_0.1.0                    shiny_1.8.0                  
FALSE  [31] GenomeInfoDbData_1.2.11       digest_0.6.34                
FALSE  [33] aplot_0.2.2                   enrichplot_1.22.0            
FALSE  [35] colorspace_2.1-0              patchwork_1.2.0              
FALSE  [37] rprojroot_2.0.4               RSQLite_2.3.5                
FALSE  [39] filelock_1.0.3                labeling_0.4.3               
FALSE  [41] fansi_1.0.6                   httr_1.4.7                   
FALSE  [43] polyclip_1.10-6               compiler_4.3.2               
FALSE  [45] bit64_4.0.5                   withr_3.0.0                  
FALSE  [47] BiocParallel_1.36.0           viridis_0.6.5                
FALSE  [49] DBI_1.2.2                     highr_0.10                   
FALSE  [51] ggforce_0.4.2                 MASS_7.3-60.0.1              
FALSE  [53] rappdirs_0.3.3                HDO.db_0.99.1                
FALSE  [55] tools_4.3.2                   interactiveDisplayBase_1.40.0
FALSE  [57] ape_5.7-1                     scatterpie_0.2.1             
FALSE  [59] httpuv_1.6.14                 glue_1.7.0                   
FALSE  [61] nlme_3.1-164                  GOSemSim_2.28.1              
FALSE  [63] promises_1.2.1                grid_4.3.2                   
FALSE  [65] shadowtext_0.1.3              Rtsne_0.17                   
FALSE  [67] reshape2_1.4.4                generics_0.1.3               
FALSE  [69] gtable_0.3.4                  tzdb_0.4.0                   
FALSE  [71] data.table_1.15.2             hms_1.1.3                    
FALSE  [73] tidygraph_1.3.1               utf8_1.2.4                   
FALSE  [75] XVector_0.42.0                ggrepel_0.9.5                
FALSE  [77] BiocVersion_3.18.1            pillar_1.9.0                 
FALSE  [79] yulab.utils_0.1.4             later_1.3.2                  
FALSE  [81] splines_4.3.2                 tweenr_2.0.3                 
FALSE  [83] BiocFileCache_2.10.1          treeio_1.26.0                
FALSE  [85] lattice_0.22-5                bit_4.0.5                    
FALSE  [87] tidyselect_1.2.0              GO.db_3.18.0                 
FALSE  [89] Biostrings_2.70.2             knitr_1.45                   
FALSE  [91] git2r_0.33.0                  gridExtra_2.3                
FALSE  [93] xfun_0.42                     graphlayouts_1.1.1           
FALSE  [95] stringi_1.8.3                 workflowr_1.7.1              
FALSE  [97] lazyeval_0.2.2                ggfun_0.1.4                  
FALSE  [99] yaml_2.3.8                    evaluate_0.23                
FALSE [101] codetools_0.2-19              ggraph_2.2.1                 
FALSE [103] tibble_3.2.1                  qvalue_2.34.0                
FALSE [105] BiocManager_1.30.22           ggplotify_0.1.2              
FALSE [107] cli_3.6.2                     xtable_1.8-4                 
FALSE [109] munsell_0.5.1                 jquerylib_0.1.4              
FALSE [111] Rcpp_1.0.12                   GenomeInfoDb_1.38.8          
FALSE [113] dbplyr_2.4.0                  png_0.1-8                    
FALSE [115] parallel_4.3.2                ellipsis_0.3.2               
FALSE [117] readr_2.1.5                   blob_1.2.4                   
FALSE [119] DOSE_3.28.2                   bitops_1.0-7                 
FALSE [121] viridisLite_0.4.2             tidytree_0.4.6               
FALSE [123] scales_1.3.0                  crayon_1.5.2                 
FALSE [125] rlang_1.1.3                   cowplot_1.1.3                
FALSE [127] fastmatch_1.1-4               KEGGREST_1.42.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
