<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Carbonetto" />


<title>The MCF-7 data set as possible model system for the Poisson log1p NMF model</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">log1p_experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eweine/log1p_experiments">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">The MCF-7 data set as possible model system
for the Poisson log1p NMF model</h1>
<h4 class="author">Peter Carbonetto</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-05-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>log1p_experiments/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240402code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240402)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240402code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240402)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstreec359b26351199a7070c0e9a7ff03d8ef8c43fcd3targetblankc359b26a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/eweine/log1p_experiments/tree/c359b26351199a7070c0e9a7ff03d8ef8c43fcd3" target="_blank">c359b26</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstreec359b26351199a7070c0e9a7ff03d8ef8c43fcd3targetblankc359b26a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/eweine/log1p_experiments/tree/c359b26351199a7070c0e9a7ff03d8ef8c43fcd3" target="_blank">c359b26</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  data/pancreas.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/mcf7.Rmd</code>) and HTML
(<code>docs/mcf7.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/e35958d1195c25efda4f6e7046c2b91203733d7b/analysis/mcf7.Rmd" target="_blank">e35958d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-30
</td>
<td>
Small fix to mcf7.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/032e50dd7d92800a1b26cbdda4d50c05f43503e3/analysis/mcf7.Rmd" target="_blank">032e50d</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-30
</td>
<td>
Added NNLM fit to mcf7.Rmd.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/6c00be7527dde9fe302be83982db4d863ab7b9b6/analysis/mcf7.Rmd" target="_blank">6c00be7</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-30
</td>
<td>
Removed ncRNA genes from analysis of the MCF-7 data.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/5511ab8603693e1500afbf26646fec18cadee6ae/analysis/mcf7.Rmd" target="_blank">5511ab8</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-27
</td>
<td>
Added fastglmpca fit to mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/e0c09df6a1c02271da9cc14b43086b952bbe85fa/analysis/mcf7.Rmd" target="_blank">e0c09df</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-25
</td>
<td>
added PC plot with fitted topics
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/e0c09df6a1c02271da9cc14b43086b952bbe85fa/docs/mcf7.html" target="_blank">e0c09df</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-25
</td>
<td>
added PC plot with fitted topics
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/5b3cd036fb148cba1ed0574bbdc79bebe008a71c/analysis/mcf7.Rmd" target="_blank">5b3cd03</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-25
</td>
<td>
added NMF fit with new initialization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/5b3cd036fb148cba1ed0574bbdc79bebe008a71c/docs/mcf7.html" target="_blank">5b3cd03</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-25
</td>
<td>
added NMF fit with new initialization
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/3f1b9b5284b78ce0576c69c78b23e3a867cae059/analysis/mcf7.Rmd" target="_blank">3f1b9b5</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-23
</td>
<td>
added log1p fits
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/3f1b9b5284b78ce0576c69c78b23e3a867cae059/docs/mcf7.html" target="_blank">3f1b9b5</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-23
</td>
<td>
added log1p fits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/bde6ab29de1f9a9d2311c2fd0fe618b6eb8b458b/analysis/mcf7.Rmd" target="_blank">bde6ab2</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-20
</td>
<td>
Added PCA to mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/95270b1558e64b876af314217cae90a803dee621/analysis/mcf7.Rmd" target="_blank">95270b1</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-11
</td>
<td>
Working on some additional analyses to add to mcf7.Rmd.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/99138245809314e367d6c0d6909fe698678b6616/docs/mcf7.html" target="_blank">9913824</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
Ran wflow_publish("analysis/mcf7.Rmd").
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/603a5aad3c0de0a582f824c727f1496738240e6f/analysis/mcf7.Rmd" target="_blank">603a5aa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
Added a few notes to the mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/d22faede353cb51f3e255df4fd5371abf55694b6/analysis/mcf7.Rmd" target="_blank">d22faed</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
Added flashier NMF fit to mcf7 analysis.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/b777168a98cb77fe5c757b658b61546a2367f007/docs/mcf7.html" target="_blank">b777168</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
Added topic model fitting and strucure plot to the mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/3b682efdd9aaaa070406b73af9796e32ed71321e/analysis/mcf7.Rmd" target="_blank">3b682ef</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
wflow_publish("analysis/mcf7.Rmd", verbose = TRUE, view = FALSE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/57d755cedce84d2f238dd720f06bbca40edc483a/docs/mcf7.html" target="_blank">57d755c</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
Updated the template for the mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/ebd0936a170c7140fa85f131f001e3b562820b0e/analysis/mcf7.Rmd" target="_blank">ebd0936</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
workflowr::wflow_publish("mcf7.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/eweine/log1p_experiments/728ac337c01f9f3fca3cde9e31f23b87e4b2f8bd/docs/mcf7.html" target="_blank">728ac33</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
First build of the mcf7 analysis.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/7b53efaa9472f74db505237864662f232de539a2/analysis/mcf7.Rmd" target="_blank">7b53efa</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
<td>
workflowr::wflow_publish("mcf7.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Here I explore the possibility of the data from the <a
href="https://doi.org/10.7554/eLife.59388">Sanford et al eLife paper</a>
as a possible “model system” for the Poisson log1p NMF model. This paper
analyzed the transcriptional response of human MCF-7 cells to retinoic
acid and TGF-<span class="math inline">\(\beta\)</span>, applied
individually and in combination.</p>
<p>The following abbreviations were used in their data: EtOH = ethanol;
RA = retinoic acid; TGFb = TGF-<span
class="math inline">\(\beta\)</span>.</p>
<p>Note that the data were downloaded from GEO, accession GSE152749.</p>
<p>Load the fastTopics and flashier packages, and a few other packages
needed to retrieve the data and create the plots.</p>
<pre class="r"><code>library(rsvd)
library(fastglmpca)
library(fastTopics)
library(flashier)
library(data.table)
library(GEOquery)
library(ggplot2)
library(cowplot)
library(log1pNMF)</code></pre>
<p>Set the seed for reproducibility:</p>
<pre class="r"><code>set.seed(1)</code></pre>
<div id="prepare-the-data-for-analysis-with-fasttopics-and-flashier"
class="section level2">
<h2>Prepare the data for analysis with fastTopics and flashier</h2>
<p>Load the gene information:</p>
<pre class="r"><code>genes &lt;- fread(&quot;data/Human.GRCh38.p13.annot.tsv.gz&quot;,
               sep = &quot;\t&quot;,header = TRUE,stringsAsFactors = FALSE)
class(genes) &lt;- &quot;data.frame&quot;
genes &lt;- genes[1:10]
genes &lt;- transform(genes,
                   GeneType = factor(GeneType),
                   Status   = factor(Status))</code></pre>
<p>Load the RNA-seq counts:</p>
<pre class="r"><code>counts &lt;- fread(&quot;data/GSE152749_raw_counts_GRCh38.p13_NCBI.tsv.gz&quot;,
                sep = &quot;\t&quot;,header = TRUE,stringsAsFactors = FALSE)
class(counts) &lt;- &quot;data.frame&quot;
rownames(counts) &lt;- counts$GeneID
counts &lt;- counts[,-1]
counts &lt;- as.matrix(counts)
storage.mode(counts) &lt;- &quot;double&quot;
counts &lt;- t(counts)
ids &lt;- rownames(counts)</code></pre>
<p>Load the sample information:</p>
<pre class="r"><code>geo &lt;- getGEO(filename = &quot;data/GSE152749_family.soft.gz&quot;)
samples &lt;- data.frame(id = names(GSMList(geo)),
                      treatment = sapply(GSMList(geo),
                                         function (x) Meta(x)$title))
samples &lt;- samples[ids,]
rownames(samples) &lt;- NULL
samples &lt;- transform(samples,
                     EtOH = grepl(&quot;EtOH&quot;,treatment,fixed = TRUE),
                     RA   = grepl(&quot;RA&quot;,treatment,fixed = TRUE),
                     TGFb = grepl(&quot;TGFb&quot;,treatment,fixed = TRUE))
samples$label                 &lt;- &quot;EtOH&quot;
samples[samples$RA,&quot;label&quot;]   &lt;- &quot;RA&quot;
samples[samples$TGFb,&quot;label&quot;] &lt;- &quot;TGFb&quot;
samples[with(samples,RA &amp; TGFb),&quot;label&quot;] &lt;- &quot;RA+TGFb&quot;
samples &lt;- transform(samples,
                     label = factor(label,c(&quot;EtOH&quot;,&quot;RA&quot;,&quot;TGFb&quot;,&quot;RA+TGFb&quot;)))</code></pre>
<p>Remove the non-protein-coding genes, and the genes that are expressed
in fewer than 4 samples:</p>
<pre class="r"><code>x &lt;- colSums(counts &gt; 0)
i &lt;- which(x &gt; 3 &amp;
           genes$GeneType == &quot;protein-coding&quot; &amp;
           genes$Status == &quot;active&quot;)
genes  &lt;- genes[i,]
counts &lt;- counts[,i]</code></pre>
<p>(It turns out that there is some structure in the non-coding RNA
genes that is is picked up by some of the methods, so to avoid this
complication I have removed these genes from the data.)</p>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>Let’s see what happens when we apply PCA to the shifted log
counts.</p>
<p>First compute the shifted log counts:</p>
<pre class="r"><code>a &lt;- 1
s &lt;- rowSums(counts)
s &lt;- s/mean(s)
Y &lt;- log1p(counts/(a*s))</code></pre>
<p>Now run PCA on the shifted log counts, and plot the first 2 PCs:</p>
<pre class="r"><code>pca &lt;- rpca(Y,k = 2,center = TRUE,scale = FALSE)
colnames(pca$x) &lt;- c(&quot;PC1&quot;,&quot;PC2&quot;)
pdat &lt;- data.frame(samples,pca$x)
ggplot(pdat,aes(x = PC1,y = PC2,color = label)) +
  geom_point() +
  scale_color_manual(values = c(&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;,
                                &quot;limegreen&quot;)) + 
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/mcf7.Rmd/pca-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca-1">
Past versions of pca-1.png
</button>
</p>
<div id="fig-pca-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/3f1b9b5284b78ce0576c69c78b23e3a867cae059/docs/figure/mcf7.Rmd/pca-1.png" target="_blank">3f1b9b5</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="fastglmpca" class="section level2">
<h2>fastglmpca</h2>
<p>Just for interest, let’s see what happens when we compute a GLM-PCA
reduced representation of the counts.</p>
<pre class="r"><code>fit_glmpca &lt;- init_glmpca_pois(t(counts),K = 2)
fit_glmpca &lt;- fit_glmpca_pois(t(counts),fit0 = fit_glmpca,verbose = FALSE)
colnames(fit_glmpca$V) &lt;- c(&quot;k1&quot;,&quot;k2&quot;)
pdat &lt;- data.frame(samples,fit_glmpca$V)
ggplot(pdat,aes(x = k1,y = k2,color = label)) +
  geom_point() +
  scale_color_manual(values = c(&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;,
                                &quot;limegreen&quot;)) + 
  theme_cowplot(font_size = 10)</code></pre>
<p><img src="figure/mcf7.Rmd/fastglmpca-1.png" width="360" style="display: block; margin: auto;" /></p>
<p>The top 2 PCs from GLM-PCA are essentially identical to the top 2 PCs
from PCA applied to the shifted log counts:</p>
<pre class="r"><code>round(cor(pca$x,fit_glmpca$V),digits = 4)
#          k1      k2
# PC1 -0.9963  0.0803
# PC2 -0.0785 -0.9941</code></pre>
</div>
<div id="topic-model" class="section level2">
<h2>Topic model</h2>
<p>Let’s now try fitting a (multinomial) topic model to these data,
which represents an <em>additive model</em> of the transcriptional
changes in response to the different exposures. After a bit of trial and
error, it turns out that 4 topics are needed to get the right
result:</p>
<pre class="r"><code>fit0 &lt;- fit_poisson_nmf(counts,k = 3,init.method = &quot;random&quot;,numiter = 50,
                        verbose = &quot;none&quot;,
                        control = list(nc = 4,extrapolate = FALSE))
fit &lt;- fit_poisson_nmf(counts,fit0 = fit0,numiter = 50,verbose = &quot;none&quot;,
                       control = list(nc = 4,extrapolate = TRUE))</code></pre>
<p>The topic model identifies one topic for the transcriptional
responses to RA, one topic for the responses TGF-<span
class="math inline">\(\beta\)</span>, and one topic for the “control” or
“baseline” (ethanol), and the response from exposure to RA + TGF-<span
class="math inline">\(\beta\)</span> is a fairly even mixture of the RA
and TGF-<span class="math inline">\(\beta\)</span> topics.</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;tomato&quot;,&quot;darkblue&quot;,&quot;dodgerblue&quot;)
p1 &lt;- structure_plot(fit,grouping = samples$label,topics = 1:3,
                     colors = topic_colors)
p1</code></pre>
<p><img src="figure/mcf7.Rmd/structure-plot-topic-model-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-topic-model-1">
Past versions of structure-plot-topic-model-1.png
</button>
</p>
<div id="fig-structure-plot-topic-model-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/3f1b9b5284b78ce0576c69c78b23e3a867cae059/docs/figure/mcf7.Rmd/structure-plot-topic-model-1.png" target="_blank">3f1b9b5</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/b777168a98cb77fe5c757b658b61546a2367f007/docs/figure/mcf7.Rmd/structure-plot-topic-model-1.png" target="_blank">b777168</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nmf-on-the-shifted-log-counts" class="section level2">
<h2>NMF on the shifted log counts</h2>
<p>Now let’s try fitting an NMF to the shifted log counts using
flashier. Implicitly, this model is a <em>multiplicative model</em> of
the transcriptional changes in response to the different exposures.</p>
<p>Next fit an NMF to the shifted log counts using flashier:</p>
<pre class="r"><code>n  &lt;- nrow(counts)
x  &lt;- rpois(1e7,1/n)
s1 &lt;- sd(log(x + 1))
fl_nmf &lt;- flash(Y,ebnm_fn = ebnm_point_exponential,var_type = 2, 
                greedy_Kmax = 4,S = s1,backfit = TRUE,verbose = 0)</code></pre>
<div class="alert alert-warning">
<p>
<strong>Warning:</strong> The above code chunk cached its results, but
it won’t be re-run if previous chunks it depends on are updated. If you
need to use caching, it is highly recommended to also set
<code>knitr::opts_chunk$set(autodep = TRUE)</code> at the top of the
file (in a chunk that is not cached). Alternatively, you can customize
the option <code>dependson</code> for each individual chunk that is
cached. Using either <code>autodep</code> or <code>dependson</code> will
remove this warning. See the
<a href="https://yihui.name/knitr/options/#cache"
>knitr cache options</a> for more details.
</p>
</div>
<p>Note that I chose “greedy_Kmax” after some trial and error and found
that 4 was a good setting.</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;olivedrab&quot;,&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;)
res &lt;- ldf(fl_nmf,type = &quot;i&quot;)
p2 &lt;- structure_plot(res$L,grouping = samples$label,topics = c(2,3,4,1),
                     colors = topic_colors)
p2</code></pre>
<p><img src="figure/mcf7.Rmd/structure-plot-nmf-1.png" width="600" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-structure-plot-nmf-1">
Past versions of structure-plot-nmf-1.png
</button>
</p>
<div id="fig-structure-plot-nmf-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/3f1b9b5284b78ce0576c69c78b23e3a867cae059/docs/figure/mcf7.Rmd/structure-plot-nmf-1.png" target="_blank">3f1b9b5</a>
</td>
<td>
Eric Weine
</td>
<td>
2025-05-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/eweine/log1p_experiments/blob/99138245809314e367d6c0d6909fe698678b6616/docs/figure/mcf7.Rmd/structure-plot-nmf-1.png" target="_blank">9913824</a>
</td>
<td>
Peter Carbonetto
</td>
<td>
2025-05-09
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Interestingly, the NMF model applied to the shifted log count
decomposes the samples in nearly the same way as the topic model, with
one key difference: it also explicitly has a <em>baseline factor</em>
(factor 1).</p>
</div>
<div id="nmf-second-attempt" class="section level2">
<h2>NMF (second attempt)</h2>
<p>Out of curiosity, I also tried running a “vanilla” NMF method on the
shifted log counts just for comparison with EBNMF:</p>
<pre class="r"><code>library(NNLM)
nmf &lt;- nnmf(Y,k = 4,loss = &quot;mse&quot;,method = &quot;scd&quot;,
            max.iter = 100,verbose = 0,n.threads = 4)</code></pre>
<p>It is actually quite interesting to see that the loadings on the
individual factors appear to be a bit “noisier”, and they are also less
sparse:</p>
<pre class="r"><code>scale_cols &lt;- function (A, b)
  t(t(A) * b)
W &lt;- nmf$W
k &lt;- ncol(W)
d &lt;- apply(W,2,max)
W &lt;- scale_cols(W,1/d)
p3 &lt;- structure_plot(W,grouping = samples$label,topics = 4:1,
                     colors = topic_colors)
p3</code></pre>
<p><img src="figure/mcf7.Rmd/structure-plot-nnlm-1.png" width="600" style="display: block; margin: auto;" /></p>
</div>
<div id="log1p-fits" class="section level2">
<h2>Log1p Fits</h2>
<p>Below I fit the log1p model for different values of c.</p>
<pre class="r"><code>cc_vec &lt;- c(1e-3, 1)

fit_list &lt;- list()

for (cc in cc_vec) {
  
  set.seed(1)
  fit_list[[as.character(cc)]] &lt;- fit_poisson_log1p_nmf(
    Y = counts,
    K = 4,
    cc = cc,
    loglik = &quot;exact&quot;,
    control = list(verbose = FALSE)
  )
  
}</code></pre>
<p>Here is the c = 1 structure plot, which looks very similar to
flashier (though is less sparse).</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;olivedrab&quot;,&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;)
normalized_structure_plot(
  fit = fit_list$`1`,
  grouping = samples$label,
  colors = topic_colors,
  topics = c(2,3,4,1)
)$plot</code></pre>
<p>The c = 1e-3 structure plot is below, which looks a bit crazy.</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;olivedrab&quot;,&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;)
normalized_structure_plot(
  fit = fit_list$`0.001`,
  grouping = samples$label,
  colors = topic_colors,
  topics = c(2,3,4,1)
)$plot</code></pre>
<pre class="r"><code>tm_rank1 &lt;- fastTopics:::fit_pnmf_rank1(counts)

init_LL &lt;- cbind(
  tm_rank1$L,
  matrix(
    data = 1e-10,
    nrow = nrow(counts),
    ncol = 3
  )
)
rownames(init_LL) &lt;- rownames(counts)

init_FF &lt;- cbind(
  tm_rank1$F,
  matrix(
    data = 1e-10,
    nrow = ncol(counts),
    ncol = 3
  )
)

rownames(init_FF) &lt;- colnames(counts)
fit0 &lt;- init_poisson_nmf(X = counts, L = init_LL, F = init_FF)

fit &lt;- fit_poisson_nmf(X = counts, fit0 = fit0)
</code></pre>
<p>Finally, I also fit fasttopics with a rank-1 initialization just to
make sure the differences weren’t being driven by initialization.</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;tomato&quot;,&quot;darkblue&quot;,&quot;gold&quot;,&quot;skyblue&quot;)
p1 &lt;- structure_plot(fit,grouping = samples$label,topics = 1:4,
                     colors = topic_colors)
p1$plot</code></pre>
</div>
<div id="topic-model-with-alternative-initialization"
class="section level2">
<h2>Topic Model with Alternative Initialization</h2>
<p>I also attempted to initialize the topic model where each sample is
loaded on the mean of only the greatest L2 norm point in PC space
corresponding to it’s treatment group.</p>
<pre class="r"><code>pdat$l2_pc &lt;- sqrt(
  pdat$PC1 ^ 2 + pdat$PC2 ^ 2
)

# representative samples:
# RA+TGFb: GSM4625622
# RA: GSM4625634
# EtOH: GSM4625648
# TGFb: GSM4625621

FF &lt;- cbind(
  counts[&quot;GSM4625622&quot;, ],
  counts[&quot;GSM4625634&quot;, ],
  counts[&quot;GSM4625648&quot;, ],
  counts[&quot;GSM4625621&quot;, ]
)

rownames(FF) &lt;- colnames(counts)

LL &lt;- cbind(
  ifelse(pdat$label == &quot;RA+TGFb&quot;, 1, 0),
  ifelse(pdat$label == &quot;RA&quot;, 1, 0),
  ifelse(pdat$label == &quot;EtOH&quot;, 1, 0),
  ifelse(pdat$label == &quot;TGFb&quot;, 1, 0)
)
rownames(LL) &lt;- rownames(counts)

fit0 &lt;- init_poisson_nmf(X = counts, F = FF, L = LL)
fit &lt;- fit_poisson_nmf(X = counts, fit0 = fit0)</code></pre>
<p>Here is the structure plot at initialization:</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;tomato&quot;,&quot;darkblue&quot;,&quot;gold&quot;,&quot;skyblue&quot;)
structure_plot(fit0, grouping = pdat$label, topics = 1:4,
                     colors = topic_colors)$plot</code></pre>
<p>And here it is after optimization:</p>
<pre class="r"><code>topic_colors &lt;- c(&quot;tomato&quot;,&quot;darkblue&quot;,&quot;gold&quot;,&quot;skyblue&quot;)

structure_plot(fit, grouping = pdat$label, topics = 1:4,
                     colors = topic_colors)$plot</code></pre>
<pre class="r"><code>mn &lt;- t(fit$F %*% diag(apply(fit$L, 2, max)))
newX &lt;- log1p(mn)

# 1) center using the training means
#    (pca$center is a length‐p vector of column means of Y)
newX_cent &lt;- sweep(newX, 2, pca$center, FUN = &quot;-&quot;)

# 2) since you had scale = FALSE, you do NOT rescale here

# 3) project onto the PC loadings
#    (pca$v is a p×k matrix of the top k loading vectors)
scores &lt;- newX_cent %*% pca$rotation</code></pre>
<pre class="r"><code>pd2 &lt;- data.frame(
  pc1 = pdat$PC1,
  pc2 = pdat$PC2,
  label = pdat$label
)

new_pd &lt;- data.frame(
  pc1 = scores[,1],
  pc2 = scores[,2],
  label = c(&quot;topic1&quot;, &quot;topic2&quot;, &quot;topic3&quot;, &quot;topic4&quot;)
)

pd2 &lt;- rbind(pd2, new_pd)
</code></pre>
<p>Here’s a PC plot with the fitted topics:</p>
<pre class="r"><code>ggplot(pd2,aes(x = pc1,y = pc2,color = label)) +
  geom_point() +
  scale_color_manual(values = c(&quot;dodgerblue&quot;,&quot;tomato&quot;,&quot;darkblue&quot;,
                                &quot;limegreen&quot;, &quot;orange&quot;, &quot;black&quot;, &quot;gold&quot;, &quot;maroon&quot;)) + 
  theme_cowplot(font_size = 10)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 4.3.3 (2024-02-29)
# Platform: aarch64-apple-darwin20 (64-bit)
# Running under: macOS 15.4.1
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
# LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# time zone: America/Chicago
# tzcode source: internal
# 
# attached base packages:
# [1] stats     graphics  grDevices utils     datasets  methods   base     
# 
# other attached packages:
#  [1] NNLM_0.4.4          log1pNMF_0.1-4      cowplot_1.1.3      
#  [4] ggplot2_3.5.0       GEOquery_2.70.0     Biobase_2.62.0     
#  [7] BiocGenerics_0.48.1 data.table_1.15.2   flashier_1.0.55    
# [10] ebnm_1.1-34         fastTopics_0.7-24   fastglmpca_0.1-108 
# [13] rsvd_1.0.5          workflowr_1.7.1    
# 
# loaded via a namespace (and not attached):
#   [1] distr_2.9.3          pbapply_1.7-2        rlang_1.1.5         
#   [4] magrittr_2.0.3       git2r_0.33.0         horseshoe_0.2.0     
#   [7] compiler_4.3.3       getPass_0.2-4        callr_3.7.5         
#  [10] vctrs_0.6.5          reshape2_1.4.4       daarem_0.7          
#  [13] quadprog_1.5-8       stringr_1.5.1        pkgconfig_2.0.3     
#  [16] crayon_1.5.2         fastmap_1.1.1        labeling_0.4.3      
#  [19] utf8_1.2.4           promises_1.2.1       rmarkdown_2.26      
#  [22] tzdb_0.4.0           ps_1.7.6             purrr_1.0.2         
#  [25] xfun_0.42            cachem_1.0.8         trust_0.1-8         
#  [28] jsonlite_1.8.8       progress_1.2.3       highr_0.10          
#  [31] later_1.3.2          irlba_2.3.5.1        parallel_4.3.3      
#  [34] prettyunits_1.2.0    R6_2.5.1             bslib_0.6.1         
#  [37] stringi_1.8.3        RColorBrewer_1.1-3   limma_3.58.1        
#  [40] SQUAREM_2021.1       jquerylib_0.1.4      Rcpp_1.0.12         
#  [43] knitr_1.45           R.utils_2.12.3       readr_2.1.5         
#  [46] httpuv_1.6.14        Matrix_1.6-5         splines_4.3.3       
#  [49] tidyselect_1.2.1     rstudioapi_0.15.0    yaml_2.3.8          
#  [52] codetools_0.2-19     processx_3.8.3       lattice_0.22-5      
#  [55] tibble_3.2.1         plyr_1.8.9           withr_3.0.2         
#  [58] evaluate_1.0.3       Rtsne_0.17           RcppParallel_5.1.10 
#  [61] xml2_1.3.6           startupmsg_0.9.6.1   pillar_1.9.0        
#  [64] whisker_0.4.1        plotly_4.10.4        softImpute_1.4-1    
#  [67] generics_0.1.3       rprojroot_2.0.4      invgamma_1.1        
#  [70] truncnorm_1.0-9      hms_1.1.3            munsell_0.5.0       
#  [73] scales_1.3.0         ashr_2.2-66          gtools_3.9.5        
#  [76] RhpcBLASctl_0.23-42  glue_1.8.0           scatterplot3d_0.3-44
#  [79] lazyeval_0.2.2       tools_4.3.3          fs_1.6.5            
#  [82] grid_4.3.3           tidyr_1.3.1          colorspace_2.1-0    
#  [85] sfsmisc_1.1-18       deconvolveR_1.2-1    cli_3.6.4           
#  [88] Polychrome_1.5.1     fansi_1.0.6          mixsqp_0.3-54       
#  [91] viridisLite_0.4.2    dplyr_1.1.4          uwot_0.2.3          
#  [94] gtable_0.3.4         R.methodsS3_1.8.2    sass_0.4.9          
#  [97] digest_0.6.34        ggrepel_0.9.5        farver_2.1.1        
# [100] htmlwidgets_1.6.4    R.oo_1.26.0          htmltools_0.5.8.1   
# [103] lifecycle_1.0.4      httr_1.4.7           statmod_1.5.0       
# [106] MASS_7.3-60.0.1</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
