<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2024-08-14" />

<title>Objective evaluation of log1p flash normal approximation</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">log1p_experiments</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/eweine/log1p_experiments">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Objective evaluation of log1p flash normal
approximation</h1>
<h4 class="date">2024-08-14</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-08-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>log1p_experiments/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240402code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240402)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240402code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240402)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/Documents/log1p_experiments/output/factor_summary_sim.rds
</td>
<td style="text-align:left;">
output/factor_summary_sim.rds
</td>
</tr>
<tr>
<td style="text-align:left;">
~/Documents/log1p_experiments/output/mse_summary_sim.rds
</td>
<td style="text-align:left;">
output/mse_summary_sim.rds
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstree7d4d328ca453aba1071d9052540dd00e6e1f37c4targetblank7d4d328a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/eweine/log1p_experiments/tree/7d4d328ca453aba1071d9052540dd00e6e1f37c4" target="_blank">7d4d328</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomeweinelog1pexperimentstree7d4d328ca453aba1071d9052540dd00e6e1f37c4targetblank7d4d328a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/eweine/log1p_experiments/tree/7d4d328ca453aba1071d9052540dd00e6e1f37c4" target="_blank">7d4d328</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory

Untracked files:
    Untracked:  analysis/new_log1p_approx_testing.Rmd
    Untracked:  output/factor_summary_sim.rds
    Untracked:  output/mse_summary_sim.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>I wanted to try out Matthew’s idea of a normal approximation to the
log1p model in the context of matrix factorization. I generated data
from the model:</p>
<p><span class="math display">\[\begin{align*}
y_{ij} &amp;\sim \text{Poisson}(\lambda_{ij}) \\
\log(1 + \lambda_{ij}) &amp;= H_{ij} \\
H &amp;= LF^{T},
\end{align*}\]</span></p>
<p>where each element of <span class="math inline">\(L\)</span> and
<span class="math inline">\(F\)</span> are generated from a mixture with
<span class="math inline">\(\frac{2}{3}\)</span> mass at <span
class="math inline">\(0\)</span> and <span
class="math inline">\(\frac{1}{3}\)</span> mass on an <span
class="math inline">\(\text{Exponential}(3)\)</span> distribution.</p>
<pre class="r"><code>library(flashier)</code></pre>
<pre><code>Loading required package: ebnm</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre class="r"><code>library(distr)</code></pre>
<pre><code>Loading required package: startupmsg</code></pre>
<pre><code>Utilities for Start-Up Messages (version 0.9.6.1)</code></pre>
<pre><code>For more information see ?&quot;startupmsg&quot;, NEWS(&quot;startupmsg&quot;)</code></pre>
<pre><code>Loading required package: sfsmisc</code></pre>
<pre><code>Object Oriented Implementation of Distributions (version 2.9.3)</code></pre>
<pre><code>Attention: Arithmetics on distribution objects are understood as operations on corresponding random variables (r.v.s); see distrARITH().
Some functions from package &#39;stats&#39; are intentionally masked ---see distrMASK().
Note that global options are controlled by distroptions() ---c.f. ?&quot;distroptions&quot;.</code></pre>
<pre><code>For more information see ?&quot;distr&quot;, NEWS(&quot;distr&quot;), as well as
  http://distr.r-forge.r-project.org/
Package &quot;distrDoc&quot; provides a vignette to this package as well as to several extension packages; try vignette(&quot;distr&quot;).</code></pre>
<pre><code>
Attaching package: &#39;distr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    df, qqplot, sd</code></pre>
<pre class="r"><code>n &lt;- 1500
p &lt;- 750
K &lt;- 5

l_dist &lt;- UnivarMixingDistribution(
  Dirac(0),
  Exp(3),
  mixCoeff = c(2/3,1/3)
)

f_dist &lt;- UnivarMixingDistribution(
  Dirac(0),
  Exp(3),
  mixCoeff = c(2/3,1/3)
)

l_sampler &lt;- distr::r(l_dist)
f_sampler &lt;- distr::r(f_dist)</code></pre>
<p>In this case, I generated datasets of <span
class="math inline">\(1500 \times 750\)</span> where the true rank of
<span class="math inline">\(H\)</span> is <span
class="math inline">\(5\)</span>. After removing rows and columns
containing all <span class="math inline">\(0\)</span>s, the data are
about <span class="math inline">\(92\%\)</span> sparse and have a
maximum value around <span class="math inline">\(300 - 500\)</span>.
This seems fairly realistic for scRNA-seq data.</p>
<p>Then, I fit the following 3 models to the data over 10 separate
simulations:</p>
<ol style="list-style-type: decimal">
<li><p>MLE of log1p Poisson matrix factorization model with K set to 10.
I used a quadratic approximation to terms in the log-likelihood
corresponding to 0 counts, as discussed previously. I used K = 10 here
because I thought it was a bit unfair to initialize the MLE at the true
rank when I’m letting the other methods naturally determine the true
number of factors. Alternatively, I could have just fixed K = 5 for all
methods. However, I was interested in comparing how many factors each
flash model ended up using.</p></li>
<li><p>non-negative flash on the log1p transformed counts, with a
constant value of S for all values. I used a greedy initialization and
then a backfit.</p></li>
<li><p>flash on the log1p transformed counts where all <span
class="math inline">\(0\)</span> counts were set to <span
class="math inline">\(-1\)</span> and S set according to Matthew’s
approximation. Using this model, I tried both (a) a greedy
initialization followed by a backfit and (b) initializing with (1)
followed by a backfit.</p></li>
</ol>
<pre class="r"><code>get_n_factors_flash &lt;- function(f) {

  n_factors &lt;- f$n_factors

  for (j in 1:n_factors) {

    if (all(f$F_pm[, j] &lt; 1e-12)) {

      n_factors &lt;- n_factors - 1

    }

  }

  return(n_factors)

}

n_sims &lt;- 10

mse_mle_vec &lt;- c()
mse_log1p_old_vec &lt;- c()
mse_log1p_new_mle_vec &lt;- c()
mse_log1p_new_greedy_vec &lt;- c()

factors_log1p_old_vec &lt;- c()
factors_log1p_new_mle_vec &lt;- c()
factors_log1p_new_greedy_vec &lt;- c()

for (i in 1:n_sims) {

  set.seed(i)

  LL &lt;- matrix(
    data = l_sampler(n * K),
    nrow = n,
    ncol = K
  )

  FF &lt;- matrix(
    data = f_sampler(p * K),
    nrow = p,
    ncol = K
  )

  Lambda &lt;- exp(tcrossprod(LL, FF)) - 1
  y &lt;- rpois(n = length(as.vector(Lambda)), lambda = as.vector(Lambda))

  Y &lt;- matrix(
    data = y,
    nrow = n,
    ncol = p
  )

  rownames(Y) &lt;- paste0(&quot;cell&quot;, 1:n)
  colnames(Y) &lt;- paste0(&quot;gene&quot;, 1:p)

  B &lt;- tcrossprod(LL, FF)

  rownames(B) &lt;- paste0(&quot;cell&quot;, 1:n)
  colnames(B) &lt;- paste0(&quot;gene&quot;, 1:p)

  Y &lt;- Y[rowSums(Y) &gt; 0, ]
  Y &lt;- Y[, colSums(Y) &gt; 0]

  B &lt;- B[rownames(B) %in% rownames(Y), ]
  B &lt;- B[, colnames(B) %in% colnames(Y)]

  Y_trans &lt;- log1p(Y)

  x  &lt;- rpois(1e7, 1/n)
  s1 &lt;- sd(log(x + 1))

  fit_flash_old &lt;- flash(
    data = as(Y_trans, &quot;sparseMatrix&quot;),
    S = s1,
    ebnm_fn = ebnm::ebnm_point_exponential,
    backfit = TRUE
  )

  library(passPCA)

  log1p_fit &lt;- fit_factor_model_log1p_quad_approx_sparse(
    Y = as(Y, &quot;sparseMatrix&quot;),
    K = 10,
    approx_range = c(0, 1.25),
    maxiter = 500
  )

  S &lt;- as.matrix(Y / ((1 + Y) ^ 2))
  S &lt;- apply(S, c(1, 2), function(x){if(x == 0){1}else{x}})

  Y_trans2 &lt;- apply(Y_trans, c(1, 2), function(x){if(x == 0){-1}else{x}})

  fit_flash_new_mle &lt;- flash_init(data = Y_trans2, S = sqrt(S))
  fit_flash_new_mle &lt;- flash_factors_init(
    flash = fit_flash_new_mle,
    init = list(
      log1p_fit$U, log1p_fit$V
    ),
    ebnm_fn = ebnm::ebnm_point_exponential
  )

  fit_flash_new_mle &lt;- flash_backfit(fit_flash_new_mle)
  fit_flash_new_greedy &lt;- flash(data = Y_trans2, S = sqrt(S), backfit = TRUE)


  B_mle &lt;- tcrossprod(log1p_fit$U, log1p_fit$V)
  B_log1p_old &lt;- tcrossprod(fit_flash_old$L_pm, fit_flash_old$F_pm)
  B_log1p_new_mle &lt;- tcrossprod(fit_flash_new_mle$L_pm, fit_flash_new_mle$F_pm)
  B_log1p_new_greedy &lt;- tcrossprod(
    fit_flash_new_greedy$L_pm,
    fit_flash_new_greedy$F_pm
  )

  n_factors_log1p_old &lt;- get_n_factors_flash(fit_flash_old)
  n_factors_log1p_new &lt;- get_n_factors_flash(fit_flash_new_mle)
  n_factors_log1p_new_greedy &lt;- get_n_factors_flash(fit_flash_new_greedy)

  mse_mle &lt;- mean((B_mle - B) ^ 2)
  mse_log1p_old &lt;- mean((B_log1p_old - B) ^ 2)
  mse_log1p_new_mle &lt;- mean((B_log1p_new_mle - B) ^ 2)
  mse_log1p_new_greedy &lt;- mean((B_log1p_new_greedy - B) ^ 2)

  mse_mle_vec &lt;- c(mse_mle_vec, mse_mle)
  mse_log1p_old_vec &lt;- c(mse_log1p_old_vec, mse_log1p_old)
  mse_log1p_new_mle_vec &lt;- c(mse_log1p_new_mle_vec, mse_log1p_new_mle)
  mse_log1p_new_greedy_vec &lt;- c(mse_log1p_new_greedy_vec, mse_log1p_new_greedy)

  factors_log1p_old_vec &lt;- c(factors_log1p_old_vec, n_factors_log1p_old)
  factors_log1p_new_mle_vec &lt;- c(factors_log1p_new_mle_vec, n_factors_log1p_new)
  factors_log1p_new_greedy_vec &lt;- c(factors_log1p_new_greedy_vec, n_factors_log1p_new_greedy)

}

# for now let&#39;s suppose that this is correct (though I&#39;m not sure it is)
# then I would like to at least objectively evaluate what I have here

factors_res_df &lt;- data.frame(
  factors = c(factors_log1p_old_vec, factors_log1p_new_mle_vec, factors_log1p_new_greedy_vec),
  method = c(rep(&quot;flash_old&quot;, 10), rep(&quot;flash_new_mle_init&quot;, 10), rep(&quot;flash_new_greedy_init&quot;, 10))
)

library(dplyr)
df_summary &lt;- factors_res_df %&gt;%
  group_by(method) %&gt;%
  summarise(
    mean_factors = mean(factors, na.rm = TRUE),
    sd_factors = sd(factors, na.rm = TRUE)
  )

mse_res_df &lt;- data.frame(
  mse = c(mse_mle_vec, mse_log1p_new_greedy_vec, mse_log1p_new_mle_vec, mse_log1p_old_vec),
  method = c(rep(&quot;MLE&quot;, 10), rep(&quot;flash_new_greedy_init&quot;, 10), rep(&quot;flash_new_mle_init&quot;, 10), rep(&quot;flash_old&quot;, 10))
)

df_summary2 &lt;- mse_res_df %&gt;%
  group_by(method) %&gt;%
  summarise(
    mean_mse = mean(mse, na.rm = TRUE),
    sd_mse = sd(mse, na.rm = TRUE)
  )</code></pre>
<pre class="r"><code>df_summary &lt;- readr::read_rds(&quot;~/Documents/log1p_experiments/output/factor_summary_sim.rds&quot;)

df_summary2 &lt;- readr::read_rds(&quot;~/Documents/log1p_experiments/output/mse_summary_sim.rds&quot;)</code></pre>
<p>First, I evaluated the MSE of estimating <span
class="math inline">\(H\)</span>. The results are below:</p>
<pre class="r"><code>knitr::kable(df_summary2)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">method</th>
<th align="right">mean_mse</th>
<th align="right">sd_mse</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">MLE</td>
<td align="right">0.0014327</td>
<td align="right">0.0000467</td>
</tr>
<tr class="even">
<td align="left">flash_new_greedy_init</td>
<td align="right">0.6794423</td>
<td align="right">0.0090802</td>
</tr>
<tr class="odd">
<td align="left">flash_new_mle_init</td>
<td align="right">0.0125228</td>
<td align="right">0.0015140</td>
</tr>
<tr class="even">
<td align="left">flash_old</td>
<td align="right">0.0056052</td>
<td align="right">0.0010766</td>
</tr>
</tbody>
</table>
<p>Here, “flash_new” refers to Matthew’s approximation he proposed
recently and “flash_old” refers to the current procedure. Here, we can
see that the MLE performs substantially better than all methods, which
is followed by the current flash procedure. The proposed flash procedure
seems to do quite poorly with a greedy initialization, where it seems to
do okay when initialized at the MLE.</p>
<p>I also computed the mean number of non-zero factors fit by each
model. The results are below:</p>
<pre class="r"><code>knitr::kable(df_summary)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">method</th>
<th align="right">mean_factors</th>
<th align="right">sd_factors</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">flash_new_greedy_init</td>
<td align="right">6.6</td>
<td align="right">1.4298407</td>
</tr>
<tr class="even">
<td align="left">flash_new_mle_init</td>
<td align="right">8.9</td>
<td align="right">0.9944289</td>
</tr>
<tr class="odd">
<td align="left">flash_old</td>
<td align="right">41.7</td>
<td align="right">10.5309069</td>
</tr>
</tbody>
</table>
<p>Here, we can see that the old flash method typically adds far too
many factors. The new method with greedy initialization seems to do the
best, where the method initialized with the MLE seems to do okay. The
MLE initialized method generally set some of the factors to 0 in the
backfit step.</p>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>While ultimately the goal of this project is to improve
interpretation of the data which generally comes from visualizations of
model fits, I think it is useful to have some objective numbers coming
from simulations. Here, it seems like the greedy initialization seems to
work quite poorly for Matthew’s new proposal. However, things seemed to
be reasonably okay with an MLE initialization. I was a bit surprised
that the MLE outperformed the other methods in estimating <span
class="math inline">\(H\)</span>, but I suppose that the other models
are mis-specified, so they face an additional disadvantage. I think it
is worth understanding qualitative differences in these fits as well,
but I will leave that for another vignette.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Ventura 13.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] distr_2.9.3        sfsmisc_1.1-18     startupmsg_0.9.6.1 flashier_1.0.7    
[5] magrittr_2.0.3     ebnm_1.1-2        

loaded via a namespace (and not attached):
 [1] gtable_0.3.5      softImpute_1.4-1  xfun_0.45         bslib_0.7.0      
 [5] ggplot2_3.5.1     lattice_0.22-6    tzdb_0.4.0        vctrs_0.6.5      
 [9] tools_4.4.0       generics_0.1.3    parallel_4.4.0    tibble_3.2.1     
[13] fansi_1.0.6       highr_0.11        pkgconfig_2.0.3   Matrix_1.7-0     
[17] SQUAREM_2021.1    lifecycle_1.0.4   truncnorm_1.0-9   compiler_4.4.0   
[21] stringr_1.5.1     git2r_0.33.0      munsell_0.5.1     httpuv_1.6.15    
[25] htmltools_0.5.8.1 sass_0.4.9        yaml_2.3.8        later_1.3.2      
[29] pillar_1.9.0      jquerylib_0.1.4   tidyr_1.3.1       MASS_7.3-61      
[33] cachem_1.1.0      trust_0.1-8       tidyselect_1.2.1  digest_0.6.36    
[37] stringi_1.8.4     dplyr_1.1.4       purrr_1.0.2       ashr_2.2-63      
[41] splines_4.4.0     rprojroot_2.0.4   fastmap_1.2.0     grid_4.4.0       
[45] colorspace_2.1-0  cli_3.6.3         invgamma_1.1      utf8_1.2.4       
[49] readr_2.1.5       scales_1.3.0      promises_1.3.0    horseshoe_0.2.0  
[53] rmarkdown_2.27    deconvolveR_1.2-1 workflowr_1.7.1   hms_1.1.3        
[57] evaluate_0.24.0   knitr_1.47        irlba_2.3.5.1     rlang_1.1.4      
[61] Rcpp_1.0.13       mixsqp_0.3-54     glue_1.7.0        rstudioapi_0.16.0
[65] jsonlite_1.8.8    R6_2.5.1          fs_1.6.4         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
