---
title: "Applying NMF models to Mouse Brain Stimulation"
output:
  workflowr::wflow_html:
    code_folding: hide
date: "2024-10-04"
---

## Introduction

Here, I analyzed the dataset from "Single-cell analysis of experience-dependent transcriptomic states in the mouse visual cortex" by Hrvatin et al. Here, brain cells in mice were sequenced after prolonged darkness (0h), exposure to light for one hour (1h), and exposure to light for four hours (4h). 

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, comment = FALSE, message = FALSE)
```

```{r}
library(dplyr)
library(Matrix)
library(fastTopics)
load("~/Downloads/mouse_brain_stim.Rdata")

cells <- cells %>% dplyr::filter(!is.na(maintype))

counts <- counts[rownames(counts) %in% cells$`...1`, ]
counts <- counts[, Matrix::colSums(counts) > 0]
counts <- counts[Matrix::rowSums(counts) > 0, ]
```

```{r, eval=FALSE}
n <- nrow(counts)
p <- ncol(counts)
K <- 15

rs <- Matrix::rowSums(counts)
s <- rs / mean(rs)

set.seed(1)
log1p_k1 <- passPCA::fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = 1,
  maxiter = 10,
  approx_range = c(0, 1.25),
  s = s
)

init_LL <- log1p_k1$U %>%
  cbind(
    matrix(
      data = rexp(
        n = n * (K - 1), rate = 15
      ),
      nrow = n,
      ncol = K - 1
    )
  )

init_FF <- log1p_k1$V %>%
  cbind(
    matrix(
      data = rexp(
        n = p * (K - 1), rate = 15
      ),
      nrow = p,
      ncol = K - 1
    )
  )

set.seed(1)
log1p_k15 <- passPCA::fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = 15,
  maxiter = 100,
  approx_range = c(0, 1.25),
  s = s,
  init_U = init_LL,
  init_V = init_FF
)
```


```{r}
normalize_bars <- function(LL) {

  max_col <- apply(LL, 2, max)
  sweep(LL, 2, max_col, FUN = "/")

}
```

## Log1p Poisson Model

```{r}
log1p_k15 <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/mouse_brain_k15.rds"
  )
```

```{r}
LL <- normalize_bars(log1p_k15$U)
```

```{r, results='hide'}
cell_stim <- paste(cells$maintype, cells$stim)

sp <- structure_plot(LL, grouping = cell_stim, gap = 25)
rownames(log1p_k15$V) <- colnames(counts)
```

```{r}
sp
```


```{r}
library(clusterProfiler)
library(fgsea)
library(AnnotationDbi)
library(org.Mm.eg.db)

get_go_terms <- function(V) {
  
  K <- ncol(V)
  
  genes_vec <- c()
  go_terms_vec <- c()
  
  for (k in 1:K) {
    
    driving_genes <- names(sort(V[,k], decreasing = TRUE))[1:20]
    
    go_result <- enrichGO(gene = driving_genes,
                          OrgDb = org.Mm.eg.db,
                          keyType = "SYMBOL",
                          ont = "BP", 
                          pAdjustMethod = "bonferroni",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.1)@result
    
    go_result <- go_result %>%
      dplyr::filter(p.adjust < .01)
    
    go_terms <- go_result$Description
    
    genes_vec <- c(genes_vec, paste(driving_genes, collapse = ", "))
    go_terms_vec <- c(go_terms_vec, paste(go_terms, collapse = ", "))
    
  }
  
  go_df <- data.frame(
    driving_genes = genes_vec,
    go_terms = go_terms_vec,
    factor = 1:K
  )
  
  return(go_df)
  
}
```

```{r}
gdf <- get_go_terms(log1p_k15$V)
knitr::kable(gdf)
```

## log1p transformation

```{r, results='hide'}
nmf_fit <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/rcppML_nmf_k15.rds"
  )

nmf_LL <- nmf_fit$w %*% diag(x = nmf_fit$d)
nmf_LL <- normalize_bars(nmf_LL)

nmf_FF <- t(diag(x = nmf_fit$d) %*% nmf_fit$h)
rownames(nmf_FF) <- colnames(counts)
sp <- structure_plot(nmf_LL, grouping = cell_stim, gap = 25)
```

```{r}
sp
```

```{r}
gdf <- get_go_terms(nmf_FF)
knitr::kable(gdf)
```


## cNMF

```{r, results='hide'}
usage_df <- readr::read_csv(
  "~/Documents/data/passPCA/mouse_brain_cnmf_usage.csv"
)

top_genes_df <- readr::read_csv(
  "~/Documents/data/passPCA/mouse_brain_cnmf_top_usage.csv"
)

top_genes_df <- top_genes_df %>%
  dplyr::select(-`...1`)

cNMF_LL <- usage_df %>%
  dplyr::select(-bc) %>%
  as.matrix()

sp <- structure_plot(cNMF_LL, grouping = cell_stim, gap = 25)
```

```{r}
sp
```


```{r}
get_go_terms_cNMF <- function(sets_df) {
  
  K <- ncol(sets_df)
  
  genes_vec <- c()
  go_terms_vec <- c()
  
  for (k in 1:K) {
    
    driving_genes <- (sets_df[[k]])[1:20]
    
    go_result <- enrichGO(gene = driving_genes,
                          OrgDb = org.Mm.eg.db,
                          keyType = "SYMBOL",
                          ont = "BP", 
                          pAdjustMethod = "bonferroni",
                          pvalueCutoff = 0.05,
                          qvalueCutoff = 0.1)@result
    
    go_result <- go_result %>%
      dplyr::filter(p.adjust < .01)
    
    go_terms <- go_result$Description
    
    genes_vec <- c(genes_vec, paste(driving_genes, collapse = ", "))
    go_terms_vec <- c(go_terms_vec, paste(go_terms, collapse = ", "))
    
  }
  
  go_df <- data.frame(
    driving_genes = genes_vec,
    go_terms = go_terms_vec,
    factor = 1:K
  )
  
  return(go_df)
  
}

gdf_cNMF <- get_go_terms_cNMF(top_genes_df)
knitr::kable(gdf_cNMF)
```

## Subsetting the data and fitting the new flash approximation

Here, I took a subset of $7.5$K cells (of a total ~ $40$K), and ran both the log1p Poisson model (with K fixed to $10$) and the new flash approximation proposed by Matthew. Below are the structure plots of the two models:

```{r, eval=FALSE}
set.seed(1)
cells <- cells %>% dplyr::filter(!is.na(maintype))


counts <- counts[rownames(counts) %in% cells$...1, ]
counts <- counts[Matrix::rowSums(counts) > 0, ]

counts <- counts[sample(x = rownames(counts), size = 7500, replace = FALSE), ]
counts <- counts[, Matrix::colSums(counts) > 0]
counts <- as(counts, "CsparseMatrix")

ff <- passPCA::run_flash_log1p_with_greedy_init(
  Y = counts,
  var_type = 2,
  greedy_Kmax = 20
)

out <- list(
  LL = ff$L_pm,
  FF = ff$F_pm
)

n <- nrow(counts)
p <- ncol(counts)
K <- 10

rs <- Matrix::rowSums(counts)
s <- rs / mean(rs)

set.seed(1)
log1p_k1 <- passPCA::fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = 1,
  maxiter = 10,
  approx_range = c(0, 1.25),
  s = s
)

init_LL <- log1p_k1$U %>%
  cbind(
    matrix(
      data = rexp(
        n = n * (K - 1), rate = 15
      ),
      nrow = n,
      ncol = K - 1
    )
  )

init_FF <- log1p_k1$V %>%
  cbind(
    matrix(
      data = rexp(
        n = p * (K - 1), rate = 15
      ),
      nrow = p,
      ncol = K - 1
    )
  )

set.seed(1)
log1p_k10 <- passPCA::fit_factor_model_log1p_quad_approx_sparse(
  Y = counts,
  K = K,
  maxiter = 100,
  approx_range = c(0, 1.25),
  s = s,
  init_U = init_LL,
  init_V = init_FF
)
```


```{r}
set.seed(1)
cells <- cells %>% dplyr::filter(!is.na(maintype))


counts <- counts[rownames(counts) %in% cells$...1, ]
#counts <- counts[, Matrix::colSums(counts) > 0]
counts <- counts[Matrix::rowSums(counts) > 0, ]

counts <- counts[sample(x = rownames(counts), size = 7500, replace = FALSE), ]
counts <- counts[, Matrix::colSums(counts) > 0]
counts <- as(counts, "CsparseMatrix")

log1p_k10 <- readr::read_rds("~/Documents/data/passPCA/experiment_results/mouse_brain_k10.rds")
cells_sub <- cells %>%
  dplyr::filter(
    `...1` %in% rownames(counts)
  )

LL <- log1p_k10$U

LL <- normalize_bars(LL)

ct <- cells_sub$maintype
names(ct) <- cells_sub$...1
ct <- ct[rownames(counts)]

#structure_plot(LL, grouping = ct)

flash_mod <- readr::read_rds("~/Documents/data/passPCA/experiment_results/mouse_brain_flash_fit_out.rds")

LL2 <- normalize_bars(flash_mod$LL)

#structure_plot(LL2, grouping = ct, gap = 25)
```

```{r, results='hide'}
sp1 <- structure_plot(LL, grouping = ct, gap = 25)
sp2 <- structure_plot(LL2, grouping = ct, gap = 25)
```

```{r}
sp1
sp2
```

I also tried first fitting a $K = 10$ model with the log1p MLE and then backfitting with flash. The results are below:

```{r, eval=FALSE}
ff <- passPCA::run_flash_log1p_with_MLE_init(
  Y = counts,
  var_type = 2,
  K = 10
)
```

```{r}
ff <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/mouse_brain_flash_fit_mle_init_out.rds"
)
LL3 <- normalize_bars(ff$LL)
```

```{r, results='hide'}
sp3 <- structure_plot(LL3, grouping = ct, gap = 25)
```

```{r}
sp3
```

Finally, I tried fitting the model with flash greedily and setting the residual variance to be $0$ for all entries. Flash ended up fitting 9 factors this time. The structure plot is below:

```{r}
ff <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/mouse_brain_flash_fit_var_type_null.rds"
)
LL4 <- normalize_bars(ff$LL)
```

```{r, results='hide'}
sp4 <- structure_plot(LL4, grouping = ct, gap = 25)
```

```{r}
sp4
```

# K = 25 Models

```{r}
log1p_pois_nmf <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/mouse_brain_k25_log1p_pois.rds"
  )

id_pois_nmf <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/mouse_brain_k25_nmf_pois.rds"
  )

log1p_frob_nmf <- readr::read_rds(
  "~/Documents/data/passPCA/experiment_results/frob_nmf_k25.rds"
)
```

```{r}
log1p_pois_LL <- normalize_bars(log1p_pois_nmf$U)
id_pois_LL <- normalize_bars(id_pois_nmf$L)
log1p_frob_LL <- normalize_bars(log1p_frob_nmf$W)

colnames(log1p_pois_LL) <- paste0("k", 1:25)
colnames(id_pois_LL) <- paste0("k", 1:25)
colnames(log1p_frob_LL) <- paste0("k", 1:25)
```

Now, I want to make the bar plots to visualize the loadings


```{r, fig.width=12, fig.height=20}
library(tidyr)
library(stringi)
library(stringr)
cell.type <- as.factor(paste0(cells$maintype, sep = "_", cells$stim))

# Downsample the number of cells and sort them using tSNE.
set.seed(8675309)
cell.idx <- numeric(0)
cell.types <- levels(cell.type)
for (i in 1:length(cell.types)) {
  which.idx <- which(cell.type == cell.types[i])
  # Downsample common cell types.
  if (length(which.idx) > 1000) {
    which.idx <- sample(which.idx, 1000)
  }
  # Don't include rare cell types.
  if (length(which.idx) > 10) {
    # Sort using tsne.
    tsne.res <- Rtsne::Rtsne(
      log1p_pois_LL[which.idx, ],
      dims = 1,
      pca = FALSE,
      normalize = FALSE,
      perplexity = min(100, floor((length(which.idx) - 1) / 3) - 1),
      theta = 0.1,
      max_iter = 1000,
      eta = 200,
      check_duplicates = FALSE
    )$Y[, 1]
    which.idx <- which.idx[order(tsne.res)]
    cell.idx <- c(cell.idx, which.idx)
  }
}

cell.type <- cell.type[cell.idx]
cell.type <- droplevels(cell.type)

log1p_pois_LL <- log1p_pois_LL[cell.idx, ]
id_pois_LL <- id_pois_LL[cell.idx, ]
log1p_frob_LL <- log1p_frob_LL[cell.idx, ]

make.heatmap.tib <- function(FF) {
  tib <- as_tibble(scale(FF, center = FALSE, scale = apply(FF, 2, max))) %>%
    mutate(Cell.type = cell.type) %>%
    arrange(Cell.type) %>%
    mutate(Cell.idx = row_number())
  
  tib <- tib %>%
    pivot_longer(
      -c(Cell.idx, Cell.type),
      names_to = "Factor",
      values_to = "Loading",
      values_drop_na = TRUE
    ) %>%
    mutate(Factor = as.numeric(str_extract(Factor, "[0-9]+")))
  
  return(tib)
}

log1p_pois_tib <- make.heatmap.tib(log1p_pois_LL)
id_pois_tib <- make.heatmap.tib(id_pois_LL)
log1p_frob_tib <- make.heatmap.tib(log1p_frob_LL)

heatmap.tib <- log1p_pois_tib %>% mutate(Method = "log1p Poisson NMF") %>%
  bind_rows(id_pois_tib %>% mutate(Method = "ID Poisson NMF")) %>%
  bind_rows(log1p_frob_tib %>% mutate(Method = "log1p Transformation Frob. NMF")) %>%
  mutate(Method = factor(Method, levels = c("log1p Poisson NMF", "ID Poisson NMF", "log1p Transformation Frob. NMF")))

tib <- heatmap.tib %>%
  group_by(Cell.type, Cell.idx) %>%
  summarize()

cell_type_breaks <- c(1, which(tib$Cell.type[-1] != tib$Cell.type[-nrow(tib)]))
label_pos <- cell_type_breaks / 2 + c(cell_type_breaks[-1], nrow(tib)) / 2

library(ggplot2)

plt <- ggplot(heatmap.tib, aes(x = Factor, y = -Cell.idx, fill = Loading)) +
  geom_tile() +
  scale_fill_gradient(low = "white", high = "firebrick") +
  labs(y = "") +
  scale_y_continuous(breaks = -label_pos,
                     minor_breaks = NULL,
                     labels = levels(cell.type)) +
  scale_x_continuous(breaks = seq(0, 30, 5)) +
  theme_minimal() +
  geom_hline(yintercept = -cell_type_breaks, size = 0.1) +
  facet_wrap(~Method, ncol = 1, axes = "all") +
  theme(legend.position = "none", 
        strip.text = element_text(size = 16)) 

plt
```

